{
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "aaaaaaaaaa"
      },
      "source": [
        "Git clone the repo and install the requirements. (ignore the pip errors about protobuf)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 1,
      "metadata": {
        "id": "bbbbbbbbbb",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "762562bd-ea9b-421c-965f-98766458b0a8"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Mounting Google Drive...\n",
            "/\n",
            "Mounted at /content/drive\n",
            "/content/drive/MyDrive\n",
            "/content/drive/MyDrive/ComfyUI\n",
            "-= Updating ComfyUI =-\n",
            "Already up to date.\n",
            "-= Install dependencies =-\n",
            "Requirement already satisfied: accelerate in /usr/local/lib/python3.11/dist-packages (1.6.0)\n",
            "Requirement already satisfied: numpy<3.0.0,>=1.17 in /usr/local/lib/python3.11/dist-packages (from accelerate) (2.0.2)\n",
            "Requirement already satisfied: packaging>=20.0 in /usr/local/lib/python3.11/dist-packages (from accelerate) (24.2)\n",
            "Requirement already satisfied: psutil in /usr/local/lib/python3.11/dist-packages (from accelerate) (5.9.5)\n",
            "Requirement already satisfied: pyyaml in /usr/local/lib/python3.11/dist-packages (from accelerate) (6.0.2)\n",
            "Requirement already satisfied: torch>=2.0.0 in /usr/local/lib/python3.11/dist-packages (from accelerate) (2.6.0+cu124)\n",
            "Requirement already satisfied: huggingface-hub>=0.21.0 in /usr/local/lib/python3.11/dist-packages (from accelerate) (0.30.2)\n",
            "Requirement already satisfied: safetensors>=0.4.3 in /usr/local/lib/python3.11/dist-packages (from accelerate) (0.5.3)\n",
            "Requirement already satisfied: filelock in /usr/local/lib/python3.11/dist-packages (from huggingface-hub>=0.21.0->accelerate) (3.18.0)\n",
            "Requirement already satisfied: fsspec>=2023.5.0 in /usr/local/lib/python3.11/dist-packages (from huggingface-hub>=0.21.0->accelerate) (2025.3.2)\n",
            "Requirement already satisfied: requests in /usr/local/lib/python3.11/dist-packages (from huggingface-hub>=0.21.0->accelerate) (2.32.3)\n",
            "Requirement already satisfied: tqdm>=4.42.1 in /usr/local/lib/python3.11/dist-packages (from huggingface-hub>=0.21.0->accelerate) (4.67.1)\n",
            "Requirement already satisfied: typing-extensions>=3.7.4.3 in /usr/local/lib/python3.11/dist-packages (from huggingface-hub>=0.21.0->accelerate) (4.13.2)\n",
            "Requirement already satisfied: networkx in /usr/local/lib/python3.11/dist-packages (from torch>=2.0.0->accelerate) (3.4.2)\n",
            "Requirement already satisfied: jinja2 in /usr/local/lib/python3.11/dist-packages (from torch>=2.0.0->accelerate) (3.1.6)\n",
            "Collecting nvidia-cuda-nvrtc-cu12==12.4.127 (from torch>=2.0.0->accelerate)\n",
            "  Downloading nvidia_cuda_nvrtc_cu12-12.4.127-py3-none-manylinux2014_x86_64.whl.metadata (1.5 kB)\n",
            "Collecting nvidia-cuda-runtime-cu12==12.4.127 (from torch>=2.0.0->accelerate)\n",
            "  Downloading nvidia_cuda_runtime_cu12-12.4.127-py3-none-manylinux2014_x86_64.whl.metadata (1.5 kB)\n",
            "Collecting nvidia-cuda-cupti-cu12==12.4.127 (from torch>=2.0.0->accelerate)\n",
            "  Downloading nvidia_cuda_cupti_cu12-12.4.127-py3-none-manylinux2014_x86_64.whl.metadata (1.6 kB)\n",
            "Collecting nvidia-cudnn-cu12==9.1.0.70 (from torch>=2.0.0->accelerate)\n",
            "  Downloading nvidia_cudnn_cu12-9.1.0.70-py3-none-manylinux2014_x86_64.whl.metadata (1.6 kB)\n",
            "Collecting nvidia-cublas-cu12==12.4.5.8 (from torch>=2.0.0->accelerate)\n",
            "  Downloading nvidia_cublas_cu12-12.4.5.8-py3-none-manylinux2014_x86_64.whl.metadata (1.5 kB)\n",
            "Collecting nvidia-cufft-cu12==11.2.1.3 (from torch>=2.0.0->accelerate)\n",
            "  Downloading nvidia_cufft_cu12-11.2.1.3-py3-none-manylinux2014_x86_64.whl.metadata (1.5 kB)\n",
            "Collecting nvidia-curand-cu12==10.3.5.147 (from torch>=2.0.0->accelerate)\n",
            "  Downloading nvidia_curand_cu12-10.3.5.147-py3-none-manylinux2014_x86_64.whl.metadata (1.5 kB)\n",
            "Collecting nvidia-cusolver-cu12==11.6.1.9 (from torch>=2.0.0->accelerate)\n",
            "  Downloading nvidia_cusolver_cu12-11.6.1.9-py3-none-manylinux2014_x86_64.whl.metadata (1.6 kB)\n",
            "Collecting nvidia-cusparse-cu12==12.3.1.170 (from torch>=2.0.0->accelerate)\n",
            "  Downloading nvidia_cusparse_cu12-12.3.1.170-py3-none-manylinux2014_x86_64.whl.metadata (1.6 kB)\n",
            "Requirement already satisfied: nvidia-cusparselt-cu12==0.6.2 in /usr/local/lib/python3.11/dist-packages (from torch>=2.0.0->accelerate) (0.6.2)\n",
            "Requirement already satisfied: nvidia-nccl-cu12==2.21.5 in /usr/local/lib/python3.11/dist-packages (from torch>=2.0.0->accelerate) (2.21.5)\n",
            "Requirement already satisfied: nvidia-nvtx-cu12==12.4.127 in /usr/local/lib/python3.11/dist-packages (from torch>=2.0.0->accelerate) (12.4.127)\n",
            "Collecting nvidia-nvjitlink-cu12==12.4.127 (from torch>=2.0.0->accelerate)\n",
            "  Downloading nvidia_nvjitlink_cu12-12.4.127-py3-none-manylinux2014_x86_64.whl.metadata (1.5 kB)\n",
            "Requirement already satisfied: triton==3.2.0 in /usr/local/lib/python3.11/dist-packages (from torch>=2.0.0->accelerate) (3.2.0)\n",
            "Requirement already satisfied: sympy==1.13.1 in /usr/local/lib/python3.11/dist-packages (from torch>=2.0.0->accelerate) (1.13.1)\n",
            "Requirement already satisfied: mpmath<1.4,>=1.1.0 in /usr/local/lib/python3.11/dist-packages (from sympy==1.13.1->torch>=2.0.0->accelerate) (1.3.0)\n",
            "Requirement already satisfied: MarkupSafe>=2.0 in /usr/local/lib/python3.11/dist-packages (from jinja2->torch>=2.0.0->accelerate) (3.0.2)\n",
            "Requirement already satisfied: charset-normalizer<4,>=2 in /usr/local/lib/python3.11/dist-packages (from requests->huggingface-hub>=0.21.0->accelerate) (3.4.1)\n",
            "Requirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.11/dist-packages (from requests->huggingface-hub>=0.21.0->accelerate) (3.10)\n",
            "Requirement already satisfied: urllib3<3,>=1.21.1 in /usr/local/lib/python3.11/dist-packages (from requests->huggingface-hub>=0.21.0->accelerate) (2.4.0)\n",
            "Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.11/dist-packages (from requests->huggingface-hub>=0.21.0->accelerate) (2025.4.26)\n",
            "Downloading nvidia_cublas_cu12-12.4.5.8-py3-none-manylinux2014_x86_64.whl (363.4 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m363.4/363.4 MB\u001b[0m \u001b[31m4.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading nvidia_cuda_cupti_cu12-12.4.127-py3-none-manylinux2014_x86_64.whl (13.8 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m13.8/13.8 MB\u001b[0m \u001b[31m108.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading nvidia_cuda_nvrtc_cu12-12.4.127-py3-none-manylinux2014_x86_64.whl (24.6 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m24.6/24.6 MB\u001b[0m \u001b[31m84.0 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading nvidia_cuda_runtime_cu12-12.4.127-py3-none-manylinux2014_x86_64.whl (883 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m883.7/883.7 kB\u001b[0m \u001b[31m58.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading nvidia_cudnn_cu12-9.1.0.70-py3-none-manylinux2014_x86_64.whl (664.8 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m664.8/664.8 MB\u001b[0m \u001b[31m2.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading nvidia_cufft_cu12-11.2.1.3-py3-none-manylinux2014_x86_64.whl (211.5 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m211.5/211.5 MB\u001b[0m \u001b[31m5.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading nvidia_curand_cu12-10.3.5.147-py3-none-manylinux2014_x86_64.whl (56.3 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m56.3/56.3 MB\u001b[0m \u001b[31m17.0 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading nvidia_cusolver_cu12-11.6.1.9-py3-none-manylinux2014_x86_64.whl (127.9 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m127.9/127.9 MB\u001b[0m \u001b[31m7.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading nvidia_cusparse_cu12-12.3.1.170-py3-none-manylinux2014_x86_64.whl (207.5 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m207.5/207.5 MB\u001b[0m \u001b[31m6.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading nvidia_nvjitlink_cu12-12.4.127-py3-none-manylinux2014_x86_64.whl (21.1 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m21.1/21.1 MB\u001b[0m \u001b[31m65.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hInstalling collected packages: nvidia-nvjitlink-cu12, nvidia-curand-cu12, nvidia-cufft-cu12, nvidia-cuda-runtime-cu12, nvidia-cuda-nvrtc-cu12, nvidia-cuda-cupti-cu12, nvidia-cublas-cu12, nvidia-cusparse-cu12, nvidia-cudnn-cu12, nvidia-cusolver-cu12\n",
            "  Attempting uninstall: nvidia-nvjitlink-cu12\n",
            "    Found existing installation: nvidia-nvjitlink-cu12 12.5.82\n",
            "    Uninstalling nvidia-nvjitlink-cu12-12.5.82:\n",
            "      Successfully uninstalled nvidia-nvjitlink-cu12-12.5.82\n",
            "  Attempting uninstall: nvidia-curand-cu12\n",
            "    Found existing installation: nvidia-curand-cu12 10.3.6.82\n",
            "    Uninstalling nvidia-curand-cu12-10.3.6.82:\n",
            "      Successfully uninstalled nvidia-curand-cu12-10.3.6.82\n",
            "  Attempting uninstall: nvidia-cufft-cu12\n",
            "    Found existing installation: nvidia-cufft-cu12 11.2.3.61\n",
            "    Uninstalling nvidia-cufft-cu12-11.2.3.61:\n",
            "      Successfully uninstalled nvidia-cufft-cu12-11.2.3.61\n",
            "  Attempting uninstall: nvidia-cuda-runtime-cu12\n",
            "    Found existing installation: nvidia-cuda-runtime-cu12 12.5.82\n",
            "    Uninstalling nvidia-cuda-runtime-cu12-12.5.82:\n",
            "      Successfully uninstalled nvidia-cuda-runtime-cu12-12.5.82\n",
            "  Attempting uninstall: nvidia-cuda-nvrtc-cu12\n",
            "    Found existing installation: nvidia-cuda-nvrtc-cu12 12.5.82\n",
            "    Uninstalling nvidia-cuda-nvrtc-cu12-12.5.82:\n",
            "      Successfully uninstalled nvidia-cuda-nvrtc-cu12-12.5.82\n",
            "  Attempting uninstall: nvidia-cuda-cupti-cu12\n",
            "    Found existing installation: nvidia-cuda-cupti-cu12 12.5.82\n",
            "    Uninstalling nvidia-cuda-cupti-cu12-12.5.82:\n",
            "      Successfully uninstalled nvidia-cuda-cupti-cu12-12.5.82\n",
            "  Attempting uninstall: nvidia-cublas-cu12\n",
            "    Found existing installation: nvidia-cublas-cu12 12.5.3.2\n",
            "    Uninstalling nvidia-cublas-cu12-12.5.3.2:\n",
            "      Successfully uninstalled nvidia-cublas-cu12-12.5.3.2\n",
            "  Attempting uninstall: nvidia-cusparse-cu12\n",
            "    Found existing installation: nvidia-cusparse-cu12 12.5.1.3\n",
            "    Uninstalling nvidia-cusparse-cu12-12.5.1.3:\n",
            "      Successfully uninstalled nvidia-cusparse-cu12-12.5.1.3\n",
            "  Attempting uninstall: nvidia-cudnn-cu12\n",
            "    Found existing installation: nvidia-cudnn-cu12 9.3.0.75\n",
            "    Uninstalling nvidia-cudnn-cu12-9.3.0.75:\n",
            "      Successfully uninstalled nvidia-cudnn-cu12-9.3.0.75\n",
            "  Attempting uninstall: nvidia-cusolver-cu12\n",
            "    Found existing installation: nvidia-cusolver-cu12 11.6.3.83\n",
            "    Uninstalling nvidia-cusolver-cu12-11.6.3.83:\n",
            "      Successfully uninstalled nvidia-cusolver-cu12-11.6.3.83\n",
            "Successfully installed nvidia-cublas-cu12-12.4.5.8 nvidia-cuda-cupti-cu12-12.4.127 nvidia-cuda-nvrtc-cu12-12.4.127 nvidia-cuda-runtime-cu12-12.4.127 nvidia-cudnn-cu12-9.1.0.70 nvidia-cufft-cu12-11.2.1.3 nvidia-curand-cu12-10.3.5.147 nvidia-cusolver-cu12-11.6.1.9 nvidia-cusparse-cu12-12.3.1.170 nvidia-nvjitlink-cu12-12.4.127\n",
            "Looking in indexes: https://download.pytorch.org/whl/cu121\n",
            "Requirement already satisfied: torch in /usr/local/lib/python3.11/dist-packages (2.6.0+cu124)\n",
            "Requirement already satisfied: torchvision in /usr/local/lib/python3.11/dist-packages (0.21.0+cu124)\n",
            "Requirement already satisfied: torchaudio in /usr/local/lib/python3.11/dist-packages (2.6.0+cu124)\n",
            "Requirement already satisfied: filelock in /usr/local/lib/python3.11/dist-packages (from torch) (3.18.0)\n",
            "Requirement already satisfied: typing-extensions>=4.10.0 in /usr/local/lib/python3.11/dist-packages (from torch) (4.13.2)\n",
            "Requirement already satisfied: networkx in /usr/local/lib/python3.11/dist-packages (from torch) (3.4.2)\n",
            "Requirement already satisfied: jinja2 in /usr/local/lib/python3.11/dist-packages (from torch) (3.1.6)\n",
            "Requirement already satisfied: fsspec in /usr/local/lib/python3.11/dist-packages (from torch) (2025.3.2)\n",
            "Requirement already satisfied: nvidia-cuda-nvrtc-cu12==12.4.127 in /usr/local/lib/python3.11/dist-packages (from torch) (12.4.127)\n",
            "Requirement already satisfied: nvidia-cuda-runtime-cu12==12.4.127 in /usr/local/lib/python3.11/dist-packages (from torch) (12.4.127)\n",
            "Requirement already satisfied: nvidia-cuda-cupti-cu12==12.4.127 in /usr/local/lib/python3.11/dist-packages (from torch) (12.4.127)\n",
            "Requirement already satisfied: nvidia-cudnn-cu12==9.1.0.70 in /usr/local/lib/python3.11/dist-packages (from torch) (9.1.0.70)\n",
            "Requirement already satisfied: nvidia-cublas-cu12==12.4.5.8 in /usr/local/lib/python3.11/dist-packages (from torch) (12.4.5.8)\n",
            "Requirement already satisfied: nvidia-cufft-cu12==11.2.1.3 in /usr/local/lib/python3.11/dist-packages (from torch) (11.2.1.3)\n",
            "Requirement already satisfied: nvidia-curand-cu12==10.3.5.147 in /usr/local/lib/python3.11/dist-packages (from torch) (10.3.5.147)\n",
            "Requirement already satisfied: nvidia-cusolver-cu12==11.6.1.9 in /usr/local/lib/python3.11/dist-packages (from torch) (11.6.1.9)\n",
            "Requirement already satisfied: nvidia-cusparse-cu12==12.3.1.170 in /usr/local/lib/python3.11/dist-packages (from torch) (12.3.1.170)\n",
            "Requirement already satisfied: nvidia-cusparselt-cu12==0.6.2 in /usr/local/lib/python3.11/dist-packages (from torch) (0.6.2)\n",
            "Requirement already satisfied: nvidia-nccl-cu12==2.21.5 in /usr/local/lib/python3.11/dist-packages (from torch) (2.21.5)\n",
            "Requirement already satisfied: nvidia-nvtx-cu12==12.4.127 in /usr/local/lib/python3.11/dist-packages (from torch) (12.4.127)\n",
            "Requirement already satisfied: nvidia-nvjitlink-cu12==12.4.127 in /usr/local/lib/python3.11/dist-packages (from torch) (12.4.127)\n",
            "Requirement already satisfied: triton==3.2.0 in /usr/local/lib/python3.11/dist-packages (from torch) (3.2.0)\n",
            "Requirement already satisfied: sympy==1.13.1 in /usr/local/lib/python3.11/dist-packages (from torch) (1.13.1)\n",
            "Requirement already satisfied: mpmath<1.4,>=1.1.0 in /usr/local/lib/python3.11/dist-packages (from sympy==1.13.1->torch) (1.3.0)\n",
            "Requirement already satisfied: numpy in /usr/local/lib/python3.11/dist-packages (from torchvision) (2.0.2)\n",
            "Requirement already satisfied: pillow!=8.3.*,>=5.3.0 in /usr/local/lib/python3.11/dist-packages (from torchvision) (11.2.1)\n",
            "Requirement already satisfied: MarkupSafe>=2.0 in /usr/local/lib/python3.11/dist-packages (from jinja2->torch) (3.0.2)\n",
            "Collecting torchsde\n",
            "  Downloading torchsde-0.2.6-py3-none-any.whl.metadata (5.3 kB)\n",
            "Requirement already satisfied: numpy>=1.19 in /usr/local/lib/python3.11/dist-packages (from torchsde) (2.0.2)\n",
            "Requirement already satisfied: scipy>=1.5 in /usr/local/lib/python3.11/dist-packages (from torchsde) (1.15.2)\n",
            "Requirement already satisfied: torch>=1.6.0 in /usr/local/lib/python3.11/dist-packages (from torchsde) (2.6.0+cu124)\n",
            "Collecting trampoline>=0.1.2 (from torchsde)\n",
            "  Downloading trampoline-0.1.2-py3-none-any.whl.metadata (10 kB)\n",
            "Requirement already satisfied: filelock in /usr/local/lib/python3.11/dist-packages (from torch>=1.6.0->torchsde) (3.18.0)\n",
            "Requirement already satisfied: typing-extensions>=4.10.0 in /usr/local/lib/python3.11/dist-packages (from torch>=1.6.0->torchsde) (4.13.2)\n",
            "Requirement already satisfied: networkx in /usr/local/lib/python3.11/dist-packages (from torch>=1.6.0->torchsde) (3.4.2)\n",
            "Requirement already satisfied: jinja2 in /usr/local/lib/python3.11/dist-packages (from torch>=1.6.0->torchsde) (3.1.6)\n",
            "Requirement already satisfied: fsspec in /usr/local/lib/python3.11/dist-packages (from torch>=1.6.0->torchsde) (2025.3.2)\n",
            "Requirement already satisfied: nvidia-cuda-nvrtc-cu12==12.4.127 in /usr/local/lib/python3.11/dist-packages (from torch>=1.6.0->torchsde) (12.4.127)\n",
            "Requirement already satisfied: nvidia-cuda-runtime-cu12==12.4.127 in /usr/local/lib/python3.11/dist-packages (from torch>=1.6.0->torchsde) (12.4.127)\n",
            "Requirement already satisfied: nvidia-cuda-cupti-cu12==12.4.127 in /usr/local/lib/python3.11/dist-packages (from torch>=1.6.0->torchsde) (12.4.127)\n",
            "Requirement already satisfied: nvidia-cudnn-cu12==9.1.0.70 in /usr/local/lib/python3.11/dist-packages (from torch>=1.6.0->torchsde) (9.1.0.70)\n",
            "Requirement already satisfied: nvidia-cublas-cu12==12.4.5.8 in /usr/local/lib/python3.11/dist-packages (from torch>=1.6.0->torchsde) (12.4.5.8)\n",
            "Requirement already satisfied: nvidia-cufft-cu12==11.2.1.3 in /usr/local/lib/python3.11/dist-packages (from torch>=1.6.0->torchsde) (11.2.1.3)\n",
            "Requirement already satisfied: nvidia-curand-cu12==10.3.5.147 in /usr/local/lib/python3.11/dist-packages (from torch>=1.6.0->torchsde) (10.3.5.147)\n",
            "Requirement already satisfied: nvidia-cusolver-cu12==11.6.1.9 in /usr/local/lib/python3.11/dist-packages (from torch>=1.6.0->torchsde) (11.6.1.9)\n",
            "Requirement already satisfied: nvidia-cusparse-cu12==12.3.1.170 in /usr/local/lib/python3.11/dist-packages (from torch>=1.6.0->torchsde) (12.3.1.170)\n",
            "Requirement already satisfied: nvidia-cusparselt-cu12==0.6.2 in /usr/local/lib/python3.11/dist-packages (from torch>=1.6.0->torchsde) (0.6.2)\n",
            "Requirement already satisfied: nvidia-nccl-cu12==2.21.5 in /usr/local/lib/python3.11/dist-packages (from torch>=1.6.0->torchsde) (2.21.5)\n",
            "Requirement already satisfied: nvidia-nvtx-cu12==12.4.127 in /usr/local/lib/python3.11/dist-packages (from torch>=1.6.0->torchsde) (12.4.127)\n",
            "Requirement already satisfied: nvidia-nvjitlink-cu12==12.4.127 in /usr/local/lib/python3.11/dist-packages (from torch>=1.6.0->torchsde) (12.4.127)\n",
            "Requirement already satisfied: triton==3.2.0 in /usr/local/lib/python3.11/dist-packages (from torch>=1.6.0->torchsde) (3.2.0)\n",
            "Requirement already satisfied: sympy==1.13.1 in /usr/local/lib/python3.11/dist-packages (from torch>=1.6.0->torchsde) (1.13.1)\n",
            "Requirement already satisfied: mpmath<1.4,>=1.1.0 in /usr/local/lib/python3.11/dist-packages (from sympy==1.13.1->torch>=1.6.0->torchsde) (1.3.0)\n",
            "Requirement already satisfied: MarkupSafe>=2.0 in /usr/local/lib/python3.11/dist-packages (from jinja2->torch>=1.6.0->torchsde) (3.0.2)\n",
            "Downloading torchsde-0.2.6-py3-none-any.whl (61 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m61.2/61.2 kB\u001b[0m \u001b[31m5.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading trampoline-0.1.2-py3-none-any.whl (5.2 kB)\n",
            "Installing collected packages: trampoline, torchsde\n",
            "Successfully installed torchsde-0.2.6 trampoline-0.1.2\n",
            "/content/drive/MyDrive/ComfyUI/custom_nodes\n",
            "-= Initial setup ComfyUI-Manager =-\n",
            "Cloning into 'ComfyUI-Manager'...\n",
            "remote: Enumerating objects: 19405, done.\u001b[K\n",
            "remote: Counting objects: 100% (3636/3636), done.\u001b[K\n",
            "remote: Compressing objects: 100% (465/465), done.\u001b[K\n",
            "remote: Total 19405 (delta 3356), reused 3182 (delta 3171), pack-reused 15769 (from 4)\u001b[K\n",
            "Receiving objects: 100% (19405/19405), 30.09 MiB | 15.75 MiB/s, done.\n",
            "Resolving deltas: 100% (14329/14329), done.\n",
            "/content/drive/MyDrive/ComfyUI/custom_nodes/ComfyUI-Manager\n",
            "Your configuration specifies to merge with the ref 'refs/heads/main'\n",
            "from the remote, but no such ref was fetched.\n",
            "/content/drive/MyDrive/ComfyUI\n",
            "-= Install custom nodes dependencies =-\n",
            "Requirement already satisfied: GitPython in /usr/local/lib/python3.11/dist-packages (3.1.44)\n",
            "Requirement already satisfied: gitdb<5,>=4.0.1 in /usr/local/lib/python3.11/dist-packages (from GitPython) (4.0.12)\n",
            "Requirement already satisfied: smmap<6,>=3.0.1 in /usr/local/lib/python3.11/dist-packages (from gitdb<5,>=4.0.1->GitPython) (5.0.2)\n",
            "\n",
            "\u001b[1;33mWARN: The `COMFYUI_PATH` environment variable is not set. Assuming \u001b[0m\n",
            "\u001b[1;33m`custom_nodes/ComfyUI-Manager/..\u001b[0m\u001b[1;33m/../\u001b[0m\u001b[1;33m` as the ComfyUI path.\u001b[0m\n",
            "--------------------------------------------------------------------------------\n",
            "--------------------\n",
            "Restoring \u001b[1m[\u001b[0m\u001b[1;36m1\u001b[0m/\u001b[1;36m2\u001b[0m\u001b[1m]\u001b[0m: \u001b[35m/content/drive/MyDrive/ComfyUI/custom_nodes/\u001b[0m\u001b[95m__pycache__\u001b[0m\n",
            "--------------------------------------------------------------------------------\n",
            "--------------------\n",
            "Restoring \u001b[1m[\u001b[0m\u001b[1;36m2\u001b[0m/\u001b[1;36m2\u001b[0m\u001b[1m]\u001b[0m: \u001b[35m/content/drive/MyDrive/ComfyUI/custom_nodes/\u001b[0m\u001b[95mComfyUI-Manager\u001b[0m\n",
            "Install: pip packages\n",
            "\n",
            "## ComfyUI-Manager: EXECUTE => \u001b[1m[\u001b[0m\u001b[32m'/usr/bin/python3'\u001b[0m, \u001b[32m'-m'\u001b[0m, \u001b[32m'pip'\u001b[0m, \u001b[32m'install'\u001b[0m, \n",
            "\u001b[32m'GitPython'\u001b[0m\u001b[1m]\u001b[0m\n",
            "Requirement already satisfied: GitPython in /usr/local/lib/python3.11/dist-packages (3.1.44)\n",
            "Requirement already satisfied: gitdb<5,>=4.0.1 in /usr/local/lib/python3.11/dist-packages (from GitPython) (4.0.12)\n",
            "Requirement already satisfied: smmap<6,>=3.0.1 in /usr/local/lib/python3.11/dist-packages (from gitdb<5,>=4.0.1->GitPython) (5.0.2)\n",
            "\n",
            "## ComfyUI-Manager: EXECUTE => \u001b[1m[\u001b[0m\u001b[32m'/usr/bin/python3'\u001b[0m, \u001b[32m'-m'\u001b[0m, \u001b[32m'pip'\u001b[0m, \u001b[32m'install'\u001b[0m, \n",
            "\u001b[32m'PyGithub'\u001b[0m\u001b[1m]\u001b[0m\n",
            "Collecting PyGithub\n",
            "  Downloading PyGithub-2.6.1-py3-none-any.whl.metadata (3.9 kB)\n",
            "Collecting pynacl>=1.4.0 (from PyGithub)\n",
            "  Downloading PyNaCl-1.5.0-cp36-abi3-manylinux_2_17_x86_64.manylinux2014_x86_64.manylinux_2_24_x86_64.whl.metadata (8.6 kB)\n",
            "Requirement already satisfied: requests>=2.14.0 in /usr/local/lib/python3.11/dist-packages (from PyGithub) (2.32.3)\n",
            "Requirement already satisfied: pyjwt>=2.4.0 in /usr/local/lib/python3.11/dist-packages (from pyjwt[crypto]>=2.4.0->PyGithub) (2.10.1)\n",
            "Requirement already satisfied: typing-extensions>=4.0.0 in /usr/local/lib/python3.11/dist-packages (from PyGithub) (4.13.2)\n",
            "Requirement already satisfied: urllib3>=1.26.0 in /usr/local/lib/python3.11/dist-packages (from PyGithub) (2.4.0)\n",
            "Requirement already satisfied: Deprecated in /usr/local/lib/python3.11/dist-packages (from PyGithub) (1.2.18)\n",
            "Requirement already satisfied: cryptography>=3.4.0 in /usr/local/lib/python3.11/dist-packages (from pyjwt[crypto]>=2.4.0->PyGithub) (43.0.3)\n",
            "Requirement already satisfied: cffi>=1.4.1 in /usr/local/lib/python3.11/dist-packages (from pynacl>=1.4.0->PyGithub) (1.17.1)\n",
            "Requirement already satisfied: charset-normalizer<4,>=2 in /usr/local/lib/python3.11/dist-packages (from requests>=2.14.0->PyGithub) (3.4.1)\n",
            "Requirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.11/dist-packages (from requests>=2.14.0->PyGithub) (3.10)\n",
            "Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.11/dist-packages (from requests>=2.14.0->PyGithub) (2025.4.26)\n",
            "Requirement already satisfied: wrapt<2,>=1.10 in /usr/local/lib/python3.11/dist-packages (from Deprecated->PyGithub) (1.17.2)\n",
            "Requirement already satisfied: pycparser in /usr/local/lib/python3.11/dist-packages (from cffi>=1.4.1->pynacl>=1.4.0->PyGithub) (2.22)\n",
            "Downloading PyGithub-2.6.1-py3-none-any.whl (410 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m410.5/410.5 kB\u001b[0m \u001b[31m15.7 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading PyNaCl-1.5.0-cp36-abi3-manylinux_2_17_x86_64.manylinux2014_x86_64.manylinux_2_24_x86_64.whl (856 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m856.7/856.7 kB\u001b[0m \u001b[31m53.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hInstalling collected packages: pynacl, PyGithub\n",
            "Successfully installed PyGithub-2.6.1 pynacl-1.5.0\n",
            "\n",
            "## ComfyUI-Manager: EXECUTE => \u001b[1m[\u001b[0m\u001b[32m'/usr/bin/python3'\u001b[0m, \u001b[32m'-m'\u001b[0m, \u001b[32m'pip'\u001b[0m, \u001b[32m'install'\u001b[0m, \n",
            "\u001b[32m'matrix-\u001b[0m\u001b[32mclient\u001b[0m\u001b[32m==0.4.0'\u001b[0m\u001b[1m]\u001b[0m\n",
            "Collecting matrix-client==0.4.0\n",
            "  Downloading matrix_client-0.4.0-py2.py3-none-any.whl.metadata (5.0 kB)\n",
            "Requirement already satisfied: requests~=2.22 in /usr/local/lib/python3.11/dist-packages (from matrix-client==0.4.0) (2.32.3)\n",
            "Collecting urllib3~=1.21 (from matrix-client==0.4.0)\n",
            "  Downloading urllib3-1.26.20-py2.py3-none-any.whl.metadata (50 kB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m50.1/50.1 kB\u001b[0m \u001b[31m3.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hRequirement already satisfied: charset-normalizer<4,>=2 in /usr/local/lib/python3.11/dist-packages (from requests~=2.22->matrix-client==0.4.0) (3.4.1)\n",
            "Requirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.11/dist-packages (from requests~=2.22->matrix-client==0.4.0) (3.10)\n",
            "Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.11/dist-packages (from requests~=2.22->matrix-client==0.4.0) (2025.4.26)\n",
            "Downloading matrix_client-0.4.0-py2.py3-none-any.whl (43 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m43.5/43.5 kB\u001b[0m \u001b[31m4.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading urllib3-1.26.20-py2.py3-none-any.whl (144 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m144.2/144.2 kB\u001b[0m \u001b[31m11.7 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hInstalling collected packages: urllib3, matrix-client\n",
            "  Attempting uninstall: urllib3\n",
            "    Found existing installation: urllib3 2.4.0\n",
            "    Uninstalling urllib3-2.4.0:\n",
            "      Successfully uninstalled urllib3-2.4.0\n",
            "Successfully installed matrix-client-0.4.0 urllib3-1.26.20\n",
            "\u001b[1m[\u001b[0mComfyUI-Manager\u001b[1m]\u001b[0m skip black listed pip installation: \u001b[32m'transformers'\u001b[0m\n",
            "\n",
            "## ComfyUI-Manager: EXECUTE => \u001b[1m[\u001b[0m\u001b[32m'/usr/bin/python3'\u001b[0m, \u001b[32m'-m'\u001b[0m, \u001b[32m'pip'\u001b[0m, \u001b[32m'install'\u001b[0m, \n",
            "\u001b[32m'huggingface-hub>0.20'\u001b[0m\u001b[1m]\u001b[0m\n",
            "Requirement already satisfied: huggingface-hub>0.20 in /usr/local/lib/python3.11/dist-packages (0.30.2)\n",
            "Requirement already satisfied: filelock in /usr/local/lib/python3.11/dist-packages (from huggingface-hub>0.20) (3.18.0)\n",
            "Requirement already satisfied: fsspec>=2023.5.0 in /usr/local/lib/python3.11/dist-packages (from huggingface-hub>0.20) (2025.3.2)\n",
            "Requirement already satisfied: packaging>=20.9 in /usr/local/lib/python3.11/dist-packages (from huggingface-hub>0.20) (24.2)\n",
            "Requirement already satisfied: pyyaml>=5.1 in /usr/local/lib/python3.11/dist-packages (from huggingface-hub>0.20) (6.0.2)\n",
            "Requirement already satisfied: requests in /usr/local/lib/python3.11/dist-packages (from huggingface-hub>0.20) (2.32.3)\n",
            "Requirement already satisfied: tqdm>=4.42.1 in /usr/local/lib/python3.11/dist-packages (from huggingface-hub>0.20) (4.67.1)\n",
            "Requirement already satisfied: typing-extensions>=3.7.4.3 in /usr/local/lib/python3.11/dist-packages (from huggingface-hub>0.20) (4.13.2)\n",
            "Requirement already satisfied: charset-normalizer<4,>=2 in /usr/local/lib/python3.11/dist-packages (from requests->huggingface-hub>0.20) (3.4.1)\n",
            "Requirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.11/dist-packages (from requests->huggingface-hub>0.20) (3.10)\n",
            "Requirement already satisfied: urllib3<3,>=1.21.1 in /usr/local/lib/python3.11/dist-packages (from requests->huggingface-hub>0.20) (1.26.20)\n",
            "Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.11/dist-packages (from requests->huggingface-hub>0.20) (2025.4.26)\n",
            "\n",
            "## ComfyUI-Manager: EXECUTE => \u001b[1m[\u001b[0m\u001b[32m'/usr/bin/python3'\u001b[0m, \u001b[32m'-m'\u001b[0m, \u001b[32m'pip'\u001b[0m, \u001b[32m'install'\u001b[0m, \n",
            "\u001b[32m'typer'\u001b[0m\u001b[1m]\u001b[0m\n",
            "Requirement already satisfied: typer in /usr/local/lib/python3.11/dist-packages (0.15.3)\n",
            "Requirement already satisfied: click>=8.0.0 in /usr/local/lib/python3.11/dist-packages (from typer) (8.1.8)\n",
            "Requirement already satisfied: typing-extensions>=3.7.4.3 in /usr/local/lib/python3.11/dist-packages (from typer) (4.13.2)\n",
            "Requirement already satisfied: shellingham>=1.3.0 in /usr/local/lib/python3.11/dist-packages (from typer) (1.5.4)\n",
            "Requirement already satisfied: rich>=10.11.0 in /usr/local/lib/python3.11/dist-packages (from typer) (13.9.4)\n",
            "Requirement already satisfied: markdown-it-py>=2.2.0 in /usr/local/lib/python3.11/dist-packages (from rich>=10.11.0->typer) (3.0.0)\n",
            "Requirement already satisfied: pygments<3.0.0,>=2.13.0 in /usr/local/lib/python3.11/dist-packages (from rich>=10.11.0->typer) (2.19.1)\n",
            "Requirement already satisfied: mdurl~=0.1 in /usr/local/lib/python3.11/dist-packages (from markdown-it-py>=2.2.0->rich>=10.11.0->typer) (0.1.2)\n",
            "\n",
            "## ComfyUI-Manager: EXECUTE => \u001b[1m[\u001b[0m\u001b[32m'/usr/bin/python3'\u001b[0m, \u001b[32m'-m'\u001b[0m, \u001b[32m'pip'\u001b[0m, \u001b[32m'install'\u001b[0m, \n",
            "\u001b[32m'rich'\u001b[0m\u001b[1m]\u001b[0m\n",
            "Requirement already satisfied: rich in /usr/local/lib/python3.11/dist-packages (13.9.4)\n",
            "Requirement already satisfied: markdown-it-py>=2.2.0 in /usr/local/lib/python3.11/dist-packages (from rich) (3.0.0)\n",
            "Requirement already satisfied: pygments<3.0.0,>=2.13.0 in /usr/local/lib/python3.11/dist-packages (from rich) (2.19.1)\n",
            "Requirement already satisfied: mdurl~=0.1 in /usr/local/lib/python3.11/dist-packages (from markdown-it-py>=2.2.0->rich) (0.1.2)\n",
            "\n",
            "## ComfyUI-Manager: EXECUTE => \u001b[1m[\u001b[0m\u001b[32m'/usr/bin/python3'\u001b[0m, \u001b[32m'-m'\u001b[0m, \u001b[32m'pip'\u001b[0m, \u001b[32m'install'\u001b[0m, \n",
            "\u001b[32m'typing-extensions'\u001b[0m\u001b[1m]\u001b[0m\n",
            "Requirement already satisfied: typing-extensions in /usr/local/lib/python3.11/dist-packages (4.13.2)\n",
            "\n",
            "## ComfyUI-Manager: EXECUTE => \u001b[1m[\u001b[0m\u001b[32m'/usr/bin/python3'\u001b[0m, \u001b[32m'-m'\u001b[0m, \u001b[32m'pip'\u001b[0m, \u001b[32m'install'\u001b[0m, \n",
            "\u001b[32m'toml'\u001b[0m\u001b[1m]\u001b[0m\n",
            "Requirement already satisfied: toml in /usr/local/lib/python3.11/dist-packages (0.10.2)\n",
            "\n",
            "## ComfyUI-Manager: EXECUTE => \u001b[1m[\u001b[0m\u001b[32m'/usr/bin/python3'\u001b[0m, \u001b[32m'-m'\u001b[0m, \u001b[32m'pip'\u001b[0m, \u001b[32m'install'\u001b[0m, \n",
            "\u001b[32m'uv'\u001b[0m\u001b[1m]\u001b[0m\n",
            "Collecting uv\n",
            "  Downloading uv-0.7.2-py3-none-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (11 kB)\n",
            "Downloading uv-0.7.2-py3-none-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (17.4 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m17.4/17.4 MB\u001b[0m \u001b[31m98.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hInstalling collected packages: uv\n",
            "Successfully installed uv-0.7.2\n",
            "\n",
            "## ComfyUI-Manager: EXECUTE => \u001b[1m[\u001b[0m\u001b[32m'/usr/bin/python3'\u001b[0m, \u001b[32m'-m'\u001b[0m, \u001b[32m'pip'\u001b[0m, \u001b[32m'install'\u001b[0m, \n",
            "\u001b[32m'chardet'\u001b[0m\u001b[1m]\u001b[0m\n",
            "Requirement already satisfied: chardet in /usr/local/lib/python3.11/dist-packages (5.2.0)\n",
            "\u001b[31mERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\n",
            "thinc 8.3.6 requires numpy<3.0.0,>=2.0.0, but you have numpy 1.26.4 which is incompatible.\u001b[0m\u001b[31m\n",
            "\u001b[0m"
          ]
        }
      ],
      "source": [
        "# #@title Environment Setup\n",
        "\n",
        "from pathlib import Path\n",
        "\n",
        "OPTIONS = {}\n",
        "\n",
        "USE_GOOGLE_DRIVE = True  #@param {type:\"boolean\"}\n",
        "UPDATE_COMFY_UI = True  #@param {type:\"boolean\"}\n",
        "USE_COMFYUI_MANAGER = True  #@param {type:\"boolean\"}\n",
        "INSTALL_CUSTOM_NODES_DEPENDENCIES = True  #@param {type:\"boolean\"}\n",
        "OPTIONS['USE_GOOGLE_DRIVE'] = USE_GOOGLE_DRIVE\n",
        "OPTIONS['UPDATE_COMFY_UI'] = UPDATE_COMFY_UI\n",
        "OPTIONS['USE_COMFYUI_MANAGER'] = USE_COMFYUI_MANAGER\n",
        "OPTIONS['INSTALL_CUSTOM_NODES_DEPENDENCIES'] = INSTALL_CUSTOM_NODES_DEPENDENCIES\n",
        "\n",
        "current_dir = !pwd\n",
        "WORKSPACE = f\"{current_dir[0]}/ComfyUI\"\n",
        "\n",
        "if OPTIONS['USE_GOOGLE_DRIVE']:\n",
        "    !echo \"Mounting Google Drive...\"\n",
        "    %cd /\n",
        "\n",
        "    from google.colab import drive\n",
        "    drive.mount('/content/drive')\n",
        "\n",
        "    WORKSPACE = \"/content/drive/MyDrive/ComfyUI\"\n",
        "    %cd /content/drive/MyDrive\n",
        "\n",
        "![ ! -d $WORKSPACE ] && echo -= Initial setup ComfyUI =- && git clone https://github.com/comfyanonymous/ComfyUI\n",
        "%cd $WORKSPACE\n",
        "\n",
        "if OPTIONS['UPDATE_COMFY_UI']:\n",
        "  !echo -= Updating ComfyUI =-\n",
        "\n",
        "  # Correction of the issue of permissions being deleted on Google Drive.\n",
        "  ![ -f \".ci/nightly/update_windows/update_comfyui_and_python_dependencies.bat\" ] && chmod 755 .ci/nightly/update_windows/update_comfyui_and_python_dependencies.bat\n",
        "  ![ -f \".ci/nightly/windows_base_files/run_nvidia_gpu.bat\" ] && chmod 755 .ci/nightly/windows_base_files/run_nvidia_gpu.bat\n",
        "  ![ -f \".ci/update_windows/update_comfyui_and_python_dependencies.bat\" ] && chmod 755 .ci/update_windows/update_comfyui_and_python_dependencies.bat\n",
        "  ![ -f \".ci/update_windows_cu118/update_comfyui_and_python_dependencies.bat\" ] && chmod 755 .ci/update_windows_cu118/update_comfyui_and_python_dependencies.bat\n",
        "  ![ -f \".ci/update_windows/update.py\" ] && chmod 755 .ci/update_windows/update.py\n",
        "  ![ -f \".ci/update_windows/update_comfyui.bat\" ] && chmod 755 .ci/update_windows/update_comfyui.bat\n",
        "  ![ -f \".ci/update_windows/README_VERY_IMPORTANT.txt\" ] && chmod 755 .ci/update_windows/README_VERY_IMPORTANT.txt\n",
        "  ![ -f \".ci/update_windows/run_cpu.bat\" ] && chmod 755 .ci/update_windows/run_cpu.bat\n",
        "  ![ -f \".ci/update_windows/run_nvidia_gpu.bat\" ] && chmod 755 .ci/update_windows/run_nvidia_gpu.bat\n",
        "\n",
        "  !git pull\n",
        "\n",
        "!echo -= Install dependencies =-\n",
        "!pip3 install accelerate\n",
        "!pip3 install einops transformers>=4.28.1 safetensors>=0.4.2 aiohttp pyyaml Pillow scipy tqdm psutil tokenizers>=0.13.3\n",
        "!pip3 install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/cu121\n",
        "!pip3 install torchsde\n",
        "!pip3 install kornia>=0.7.1 spandrel soundfile sentencepiece\n",
        "\n",
        "if OPTIONS['USE_COMFYUI_MANAGER']:\n",
        "  %cd custom_nodes\n",
        "\n",
        "  # Correction of the issue of permissions being deleted on Google Drive.\n",
        "  ![ -f \"ComfyUI-Manager/check.sh\" ] && chmod 755 ComfyUI-Manager/check.sh\n",
        "  ![ -f \"ComfyUI-Manager/scan.sh\" ] && chmod 755 ComfyUI-Manager/scan.sh\n",
        "  ![ -f \"ComfyUI-Manager/node_db/dev/scan.sh\" ] && chmod 755 ComfyUI-Manager/node_db/dev/scan.sh\n",
        "  ![ -f \"ComfyUI-Manager/node_db/tutorial/scan.sh\" ] && chmod 755 ComfyUI-Manager/node_db/tutorial/scan.sh\n",
        "  ![ -f \"ComfyUI-Manager/scripts/install-comfyui-venv-linux.sh\" ] && chmod 755 ComfyUI-Manager/scripts/install-comfyui-venv-linux.sh\n",
        "  ![ -f \"ComfyUI-Manager/scripts/install-comfyui-venv-win.bat\" ] && chmod 755 ComfyUI-Manager/scripts/install-comfyui-venv-win.bat\n",
        "\n",
        "  ![ ! -d ComfyUI-Manager ] && echo -= Initial setup ComfyUI-Manager =- && git clone https://github.com/ltdrdata/ComfyUI-Manager\n",
        "  %cd ComfyUI-Manager\n",
        "  !git pull\n",
        "\n",
        "%cd $WORKSPACE\n",
        "\n",
        "if OPTIONS['INSTALL_CUSTOM_NODES_DEPENDENCIES']:\n",
        "  !echo -= Install custom nodes dependencies =-\n",
        "  !pip install GitPython\n",
        "  !python custom_nodes/ComfyUI-Manager/cm-cli.py restore-dependencies\n"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "cccccccccc"
      },
      "source": [
        "Download some models/checkpoints/vae or custom comfyui nodes (uncomment the commands for the ones you want)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 2,
      "metadata": {
        "id": "dddddddddd",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "734fe536-0539-4543-fcdf-1898fbd24ad4"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2025-05-07 03:40:31--  https://huggingface.co/runwayml/stable-diffusion-v1-5/resolve/main/v1-5-pruned-emaonly.ckpt\n",
            "Resolving huggingface.co (huggingface.co)... 18.239.50.80, 18.239.50.103, 18.239.50.16, ...\n",
            "Connecting to huggingface.co (huggingface.co)|18.239.50.80|:443... connected.\n",
            "HTTP request sent, awaiting response... 307 Temporary Redirect\n",
            "Location: /stable-diffusion-v1-5/stable-diffusion-v1-5/resolve/main/v1-5-pruned-emaonly.ckpt [following]\n",
            "--2025-05-07 03:40:31--  https://huggingface.co/stable-diffusion-v1-5/stable-diffusion-v1-5/resolve/main/v1-5-pruned-emaonly.ckpt\n",
            "Reusing existing connection to huggingface.co:443.\n",
            "HTTP request sent, awaiting response... 302 Found\n",
            "Location: https://cdn-lfs.hf.co/repos/66/6f/666f465fa70158515404e8de2c6bc6fe2f90c46f9296293aa14daededeb32c52/cc6cb27103417325ff94f52b7a5d2dde45a7515b25c255d8e396c90014281516?response-content-disposition=inline%3B+filename*%3DUTF-8%27%27v1-5-pruned-emaonly.ckpt%3B+filename%3D%22v1-5-pruned-emaonly.ckpt%22%3B&Expires=1746591657&Policy=eyJTdGF0ZW1lbnQiOlt7IkNvbmRpdGlvbiI6eyJEYXRlTGVzc1RoYW4iOnsiQVdTOkVwb2NoVGltZSI6MTc0NjU5MTY1N319LCJSZXNvdXJjZSI6Imh0dHBzOi8vY2RuLWxmcy5oZi5jby9yZXBvcy82Ni82Zi82NjZmNDY1ZmE3MDE1ODUxNTQwNGU4ZGUyYzZiYzZmZTJmOTBjNDZmOTI5NjI5M2FhMTRkYWVkZWRlYjMyYzUyL2NjNmNiMjcxMDM0MTczMjVmZjk0ZjUyYjdhNWQyZGRlNDVhNzUxNWIyNWMyNTVkOGUzOTZjOTAwMTQyODE1MTY%7EcmVzcG9uc2UtY29udGVudC1kaXNwb3NpdGlvbj0qIn1dfQ__&Signature=p-t7mTFHiVMN4IxfKNGq2Lh2BxCsx8e0gMpfdxGj5oIP%7E8CJLByJWgZItVSrW30zDvnzD9xDILJK5MjdJpBwaMK3Lfl1nDgNVHiGbuBAcv6sCORwwMJ8tI-XXKbafSA4%7E-yLZU%7EZYm0Iu9HFKec8UPJS7CajjQuSa4b81cJwTmMfyrhrdUaKBeYHBVEhJr5Z0PlHUmHysGwio1cd8ojbz9QdrOBy22ZEdc9HzG4VR9qVv-RJ6H5o6q43qEN-6Ltd780I1jhVurTpmhwvQBaSRfCwomCJW2Xxm6lm49huTc2Y%7EZWWKd6BJWgA5kojKsVDoB5HOgEpJbjhS%7EKQtFfXTQ__&Key-Pair-Id=K3RPWS32NSSJCE [following]\n",
            "--2025-05-07 03:40:31--  https://cdn-lfs.hf.co/repos/66/6f/666f465fa70158515404e8de2c6bc6fe2f90c46f9296293aa14daededeb32c52/cc6cb27103417325ff94f52b7a5d2dde45a7515b25c255d8e396c90014281516?response-content-disposition=inline%3B+filename*%3DUTF-8%27%27v1-5-pruned-emaonly.ckpt%3B+filename%3D%22v1-5-pruned-emaonly.ckpt%22%3B&Expires=1746591657&Policy=eyJTdGF0ZW1lbnQiOlt7IkNvbmRpdGlvbiI6eyJEYXRlTGVzc1RoYW4iOnsiQVdTOkVwb2NoVGltZSI6MTc0NjU5MTY1N319LCJSZXNvdXJjZSI6Imh0dHBzOi8vY2RuLWxmcy5oZi5jby9yZXBvcy82Ni82Zi82NjZmNDY1ZmE3MDE1ODUxNTQwNGU4ZGUyYzZiYzZmZTJmOTBjNDZmOTI5NjI5M2FhMTRkYWVkZWRlYjMyYzUyL2NjNmNiMjcxMDM0MTczMjVmZjk0ZjUyYjdhNWQyZGRlNDVhNzUxNWIyNWMyNTVkOGUzOTZjOTAwMTQyODE1MTY%7EcmVzcG9uc2UtY29udGVudC1kaXNwb3NpdGlvbj0qIn1dfQ__&Signature=p-t7mTFHiVMN4IxfKNGq2Lh2BxCsx8e0gMpfdxGj5oIP%7E8CJLByJWgZItVSrW30zDvnzD9xDILJK5MjdJpBwaMK3Lfl1nDgNVHiGbuBAcv6sCORwwMJ8tI-XXKbafSA4%7E-yLZU%7EZYm0Iu9HFKec8UPJS7CajjQuSa4b81cJwTmMfyrhrdUaKBeYHBVEhJr5Z0PlHUmHysGwio1cd8ojbz9QdrOBy22ZEdc9HzG4VR9qVv-RJ6H5o6q43qEN-6Ltd780I1jhVurTpmhwvQBaSRfCwomCJW2Xxm6lm49huTc2Y%7EZWWKd6BJWgA5kojKsVDoB5HOgEpJbjhS%7EKQtFfXTQ__&Key-Pair-Id=K3RPWS32NSSJCE\n",
            "Resolving cdn-lfs.hf.co (cdn-lfs.hf.co)... 18.65.39.31, 18.65.39.123, 18.65.39.121, ...\n",
            "Connecting to cdn-lfs.hf.co (cdn-lfs.hf.co)|18.65.39.31|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 4265380512 (4.0G) [binary/octet-stream]\n",
            "Saving to: ‘./models/checkpoints/v1-5-pruned-emaonly.ckpt’\n",
            "\n",
            "v1-5-pruned-emaonly 100%[===================>]   3.97G  61.3MB/s    in 66s     \n",
            "\n",
            "2025-05-07 03:41:37 (61.4 MB/s) - ‘./models/checkpoints/v1-5-pruned-emaonly.ckpt’ saved [4265380512/4265380512]\n",
            "\n",
            "--2025-05-07 03:41:38--  https://huggingface.co/stabilityai/sd-vae-ft-mse-original/resolve/main/vae-ft-mse-840000-ema-pruned.safetensors\n",
            "Resolving huggingface.co (huggingface.co)... 18.239.50.16, 18.239.50.80, 18.239.50.49, ...\n",
            "Connecting to huggingface.co (huggingface.co)|18.239.50.16|:443... connected.\n",
            "HTTP request sent, awaiting response... 302 Found\n",
            "Location: https://cdn-lfs.hf.co/repos/ec/ee/eceee26c5834d8a75cf04eeb17dfc06d1d5fe1d80c2f19520b148c11e2e98c45/735e4c3a447a3255760d7f86845f09f937809baa529c17370d83e4c3758f3c75?response-content-disposition=inline%3B+filename*%3DUTF-8%27%27vae-ft-mse-840000-ema-pruned.safetensors%3B+filename%3D%22vae-ft-mse-840000-ema-pruned.safetensors%22%3B&Expires=1746592898&Policy=eyJTdGF0ZW1lbnQiOlt7IkNvbmRpdGlvbiI6eyJEYXRlTGVzc1RoYW4iOnsiQVdTOkVwb2NoVGltZSI6MTc0NjU5Mjg5OH19LCJSZXNvdXJjZSI6Imh0dHBzOi8vY2RuLWxmcy5oZi5jby9yZXBvcy9lYy9lZS9lY2VlZTI2YzU4MzRkOGE3NWNmMDRlZWIxN2RmYzA2ZDFkNWZlMWQ4MGMyZjE5NTIwYjE0OGMxMWUyZTk4YzQ1LzczNWU0YzNhNDQ3YTMyNTU3NjBkN2Y4Njg0NWYwOWY5Mzc4MDliYWE1MjljMTczNzBkODNlNGMzNzU4ZjNjNzU%7EcmVzcG9uc2UtY29udGVudC1kaXNwb3NpdGlvbj0qIn1dfQ__&Signature=M8F3t5a6lPt70a7WbnkguaRCUnGF6uvVWnOlnr%7Eb6vG1rjOl%7E6SIXsY3OHXCKdbQP32lXmfuKPDZvr6sWFOXrkEEso35bfmh6Ut7BtuwEfWPftf-SfqgvXZ3qjMTTKzkwz0Kw8UnL6cHLYwGAs5DDSZnB1pPXDObXjzi1eBUPGArBJn2Odxd0vEAaetj4YepfHlKD8pME6br5%7EqLT7W0LNl6rmvAEjBaMzOWV9a43Cw%7EH5lQKq3pjvLmuyC58yfx4guiUwD9454BJrEDKSAnba6vhgeaKuK%7EHdfvUgGUrFC99c5sp%7EbEVDqPgcrnFfYuafDpJynNwWKgMjLM-vazIQ__&Key-Pair-Id=K3RPWS32NSSJCE [following]\n",
            "--2025-05-07 03:41:38--  https://cdn-lfs.hf.co/repos/ec/ee/eceee26c5834d8a75cf04eeb17dfc06d1d5fe1d80c2f19520b148c11e2e98c45/735e4c3a447a3255760d7f86845f09f937809baa529c17370d83e4c3758f3c75?response-content-disposition=inline%3B+filename*%3DUTF-8%27%27vae-ft-mse-840000-ema-pruned.safetensors%3B+filename%3D%22vae-ft-mse-840000-ema-pruned.safetensors%22%3B&Expires=1746592898&Policy=eyJTdGF0ZW1lbnQiOlt7IkNvbmRpdGlvbiI6eyJEYXRlTGVzc1RoYW4iOnsiQVdTOkVwb2NoVGltZSI6MTc0NjU5Mjg5OH19LCJSZXNvdXJjZSI6Imh0dHBzOi8vY2RuLWxmcy5oZi5jby9yZXBvcy9lYy9lZS9lY2VlZTI2YzU4MzRkOGE3NWNmMDRlZWIxN2RmYzA2ZDFkNWZlMWQ4MGMyZjE5NTIwYjE0OGMxMWUyZTk4YzQ1LzczNWU0YzNhNDQ3YTMyNTU3NjBkN2Y4Njg0NWYwOWY5Mzc4MDliYWE1MjljMTczNzBkODNlNGMzNzU4ZjNjNzU%7EcmVzcG9uc2UtY29udGVudC1kaXNwb3NpdGlvbj0qIn1dfQ__&Signature=M8F3t5a6lPt70a7WbnkguaRCUnGF6uvVWnOlnr%7Eb6vG1rjOl%7E6SIXsY3OHXCKdbQP32lXmfuKPDZvr6sWFOXrkEEso35bfmh6Ut7BtuwEfWPftf-SfqgvXZ3qjMTTKzkwz0Kw8UnL6cHLYwGAs5DDSZnB1pPXDObXjzi1eBUPGArBJn2Odxd0vEAaetj4YepfHlKD8pME6br5%7EqLT7W0LNl6rmvAEjBaMzOWV9a43Cw%7EH5lQKq3pjvLmuyC58yfx4guiUwD9454BJrEDKSAnba6vhgeaKuK%7EHdfvUgGUrFC99c5sp%7EbEVDqPgcrnFfYuafDpJynNwWKgMjLM-vazIQ__&Key-Pair-Id=K3RPWS32NSSJCE\n",
            "Resolving cdn-lfs.hf.co (cdn-lfs.hf.co)... 18.65.39.31, 18.65.39.123, 18.65.39.121, ...\n",
            "Connecting to cdn-lfs.hf.co (cdn-lfs.hf.co)|18.65.39.31|:443... connected.\n",
            "HTTP request sent, awaiting response... 416 Requested Range Not Satisfiable\n",
            "\n",
            "    The file is already fully retrieved; nothing to do.\n",
            "\n"
          ]
        }
      ],
      "source": [
        "# Checkpoints\n",
        "\n",
        "### SDXL\n",
        "### I recommend these workflow examples: https://comfyanonymous.github.io/ComfyUI_examples/sdxl/\n",
        "\n",
        "#!wget -c https://huggingface.co/stabilityai/stable-diffusion-xl-base-1.0/resolve/main/sd_xl_base_1.0.safetensors -P ./models/checkpoints/\n",
        "#!wget -c https://huggingface.co/stabilityai/stable-diffusion-xl-refiner-1.0/resolve/main/sd_xl_refiner_1.0.safetensors -P ./models/checkpoints/\n",
        "\n",
        "# SDXL ReVision\n",
        "#!wget -c https://huggingface.co/comfyanonymous/clip_vision_g/resolve/main/clip_vision_g.safetensors -P ./models/clip_vision/\n",
        "\n",
        "# SD1.5\n",
        "!wget -c https://huggingface.co/runwayml/stable-diffusion-v1-5/resolve/main/v1-5-pruned-emaonly.ckpt -P ./models/checkpoints/\n",
        "\n",
        "# SD2\n",
        "#!wget -c https://huggingface.co/stabilityai/stable-diffusion-2-1-base/resolve/main/v2-1_512-ema-pruned.safetensors -P ./models/checkpoints/\n",
        "#!wget -c https://huggingface.co/stabilityai/stable-diffusion-2-1/resolve/main/v2-1_768-ema-pruned.safetensors -P ./models/checkpoints/\n",
        "\n",
        "# Some SD1.5 anime style\n",
        "#!wget -c https://huggingface.co/WarriorMama777/OrangeMixs/resolve/main/Models/AbyssOrangeMix2/AbyssOrangeMix2_hard.safetensors -P ./models/checkpoints/\n",
        "#!wget -c https://huggingface.co/WarriorMama777/OrangeMixs/resolve/main/Models/AbyssOrangeMix3/AOM3A1_orangemixs.safetensors -P ./models/checkpoints/\n",
        "#!wget -c https://huggingface.co/WarriorMama777/OrangeMixs/resolve/main/Models/AbyssOrangeMix3/AOM3A3_orangemixs.safetensors -P ./models/checkpoints/\n",
        "#!wget -c https://huggingface.co/Linaqruf/anything-v3.0/resolve/main/anything-v3-fp16-pruned.safetensors -P ./models/checkpoints/\n",
        "\n",
        "# Waifu Diffusion 1.5 (anime style SD2.x 768-v)\n",
        "#!wget -c https://huggingface.co/waifu-diffusion/wd-1-5-beta3/resolve/main/wd-illusion-fp16.safetensors -P ./models/checkpoints/\n",
        "\n",
        "\n",
        "# unCLIP models\n",
        "#!wget -c https://huggingface.co/comfyanonymous/illuminatiDiffusionV1_v11_unCLIP/resolve/main/illuminatiDiffusionV1_v11-unclip-h-fp16.safetensors -P ./models/checkpoints/\n",
        "#!wget -c https://huggingface.co/comfyanonymous/wd-1.5-beta2_unCLIP/resolve/main/wd-1-5-beta2-aesthetic-unclip-h-fp16.safetensors -P ./models/checkpoints/\n",
        "\n",
        "\n",
        "# VAE\n",
        "!wget -c https://huggingface.co/stabilityai/sd-vae-ft-mse-original/resolve/main/vae-ft-mse-840000-ema-pruned.safetensors -P ./models/vae/\n",
        "#!wget -c https://huggingface.co/WarriorMama777/OrangeMixs/resolve/main/VAEs/orangemix.vae.pt -P ./models/vae/\n",
        "#!wget -c https://huggingface.co/hakurei/waifu-diffusion-v1-4/resolve/main/vae/kl-f8-anime2.ckpt -P ./models/vae/\n",
        "\n",
        "\n",
        "# Loras\n",
        "#!wget -c https://civitai.com/api/download/models/10350 -O ./models/loras/theovercomer8sContrastFix_sd21768.safetensors #theovercomer8sContrastFix SD2.x 768-v\n",
        "#!wget -c https://civitai.com/api/download/models/10638 -O ./models/loras/theovercomer8sContrastFix_sd15.safetensors #theovercomer8sContrastFix SD1.x\n",
        "#!wget -c https://huggingface.co/stabilityai/stable-diffusion-xl-base-1.0/resolve/main/sd_xl_offset_example-lora_1.0.safetensors -P ./models/loras/ #SDXL offset noise lora\n",
        "\n",
        "\n",
        "# T2I-Adapter\n",
        "#!wget -c https://huggingface.co/TencentARC/T2I-Adapter/resolve/main/models/t2iadapter_depth_sd14v1.pth -P ./models/controlnet/\n",
        "#!wget -c https://huggingface.co/TencentARC/T2I-Adapter/resolve/main/models/t2iadapter_seg_sd14v1.pth -P ./models/controlnet/\n",
        "#!wget -c https://huggingface.co/TencentARC/T2I-Adapter/resolve/main/models/t2iadapter_sketch_sd14v1.pth -P ./models/controlnet/\n",
        "#!wget -c https://huggingface.co/TencentARC/T2I-Adapter/resolve/main/models/t2iadapter_keypose_sd14v1.pth -P ./models/controlnet/\n",
        "#!wget -c https://huggingface.co/TencentARC/T2I-Adapter/resolve/main/models/t2iadapter_openpose_sd14v1.pth -P ./models/controlnet/\n",
        "#!wget -c https://huggingface.co/TencentARC/T2I-Adapter/resolve/main/models/t2iadapter_color_sd14v1.pth -P ./models/controlnet/\n",
        "#!wget -c https://huggingface.co/TencentARC/T2I-Adapter/resolve/main/models/t2iadapter_canny_sd14v1.pth -P ./models/controlnet/\n",
        "\n",
        "# T2I Styles Model\n",
        "#!wget -c https://huggingface.co/TencentARC/T2I-Adapter/resolve/main/models/t2iadapter_style_sd14v1.pth -P ./models/style_models/\n",
        "\n",
        "# CLIPVision model (needed for styles model)\n",
        "#!wget -c https://huggingface.co/openai/clip-vit-large-patch14/resolve/main/pytorch_model.bin -O ./models/clip_vision/clip_vit14.bin\n",
        "\n",
        "\n",
        "# ControlNet\n",
        "#!wget -c https://huggingface.co/comfyanonymous/ControlNet-v1-1_fp16_safetensors/resolve/main/control_v11e_sd15_ip2p_fp16.safetensors -P ./models/controlnet/\n",
        "#!wget -c https://huggingface.co/comfyanonymous/ControlNet-v1-1_fp16_safetensors/resolve/main/control_v11e_sd15_shuffle_fp16.safetensors -P ./models/controlnet/\n",
        "#!wget -c https://huggingface.co/comfyanonymous/ControlNet-v1-1_fp16_safetensors/resolve/main/control_v11p_sd15_canny_fp16.safetensors -P ./models/controlnet/\n",
        "#!wget -c https://huggingface.co/comfyanonymous/ControlNet-v1-1_fp16_safetensors/resolve/main/control_v11f1p_sd15_depth_fp16.safetensors -P ./models/controlnet/\n",
        "#!wget -c https://huggingface.co/comfyanonymous/ControlNet-v1-1_fp16_safetensors/resolve/main/control_v11p_sd15_inpaint_fp16.safetensors -P ./models/controlnet/\n",
        "#!wget -c https://huggingface.co/comfyanonymous/ControlNet-v1-1_fp16_safetensors/resolve/main/control_v11p_sd15_lineart_fp16.safetensors -P ./models/controlnet/\n",
        "#!wget -c https://huggingface.co/comfyanonymous/ControlNet-v1-1_fp16_safetensors/resolve/main/control_v11p_sd15_mlsd_fp16.safetensors -P ./models/controlnet/\n",
        "#!wget -c https://huggingface.co/comfyanonymous/ControlNet-v1-1_fp16_safetensors/resolve/main/control_v11p_sd15_normalbae_fp16.safetensors -P ./models/controlnet/\n",
        "#!wget -c https://huggingface.co/comfyanonymous/ControlNet-v1-1_fp16_safetensors/resolve/main/control_v11p_sd15_openpose_fp16.safetensors -P ./models/controlnet/\n",
        "#!wget -c https://huggingface.co/comfyanonymous/ControlNet-v1-1_fp16_safetensors/resolve/main/control_v11p_sd15_scribble_fp16.safetensors -P ./models/controlnet/\n",
        "#!wget -c https://huggingface.co/comfyanonymous/ControlNet-v1-1_fp16_safetensors/resolve/main/control_v11p_sd15_seg_fp16.safetensors -P ./models/controlnet/\n",
        "#!wget -c https://huggingface.co/comfyanonymous/ControlNet-v1-1_fp16_safetensors/resolve/main/control_v11p_sd15_softedge_fp16.safetensors -P ./models/controlnet/\n",
        "#!wget -c https://huggingface.co/comfyanonymous/ControlNet-v1-1_fp16_safetensors/resolve/main/control_v11p_sd15s2_lineart_anime_fp16.safetensors -P ./models/controlnet/\n",
        "#!wget -c https://huggingface.co/comfyanonymous/ControlNet-v1-1_fp16_safetensors/resolve/main/control_v11u_sd15_tile_fp16.safetensors -P ./models/controlnet/\n",
        "\n",
        "# ControlNet SDXL\n",
        "#!wget -c https://huggingface.co/stabilityai/control-lora/resolve/main/control-LoRAs-rank256/control-lora-canny-rank256.safetensors -P ./models/controlnet/\n",
        "#!wget -c https://huggingface.co/stabilityai/control-lora/resolve/main/control-LoRAs-rank256/control-lora-depth-rank256.safetensors -P ./models/controlnet/\n",
        "#!wget -c https://huggingface.co/stabilityai/control-lora/resolve/main/control-LoRAs-rank256/control-lora-recolor-rank256.safetensors -P ./models/controlnet/\n",
        "#!wget -c https://huggingface.co/stabilityai/control-lora/resolve/main/control-LoRAs-rank256/control-lora-sketch-rank256.safetensors -P ./models/controlnet/\n",
        "\n",
        "# Controlnet Preprocessor nodes by Fannovel16\n",
        "#!cd custom_nodes && git clone https://github.com/Fannovel16/comfy_controlnet_preprocessors; cd comfy_controlnet_preprocessors && python install.py\n",
        "\n",
        "\n",
        "# GLIGEN\n",
        "#!wget -c https://huggingface.co/comfyanonymous/GLIGEN_pruned_safetensors/resolve/main/gligen_sd14_textbox_pruned_fp16.safetensors -P ./models/gligen/\n",
        "\n",
        "\n",
        "# ESRGAN upscale model\n",
        "#!wget -c https://github.com/xinntao/Real-ESRGAN/releases/download/v0.1.0/RealESRGAN_x4plus.pth -P ./models/upscale_models/\n",
        "#!wget -c https://huggingface.co/sberbank-ai/Real-ESRGAN/resolve/main/RealESRGAN_x2.pth -P ./models/upscale_models/\n",
        "#!wget -c https://huggingface.co/sberbank-ai/Real-ESRGAN/resolve/main/RealESRGAN_x4.pth -P ./models/upscale_models/\n",
        "\n",
        "\n"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "kkkkkkkkkkkkkkk"
      },
      "source": [
        "### Run ComfyUI with cloudflared (Recommended Way)\n",
        "\n",
        "\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "jjjjjjjjjjjjjj",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "4eb54708-a4af-4608-f62e-efcb899ce878"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2025-05-07 03:42:29--  https://github.com/cloudflare/cloudflared/releases/latest/download/cloudflared-linux-amd64.deb\n",
            "Resolving github.com (github.com)... 140.82.121.4\n",
            "Connecting to github.com (github.com)|140.82.121.4|:443... connected.\n",
            "HTTP request sent, awaiting response... 302 Found\n",
            "Location: https://github.com/cloudflare/cloudflared/releases/download/2025.4.2/cloudflared-linux-amd64.deb [following]\n",
            "--2025-05-07 03:42:29--  https://github.com/cloudflare/cloudflared/releases/download/2025.4.2/cloudflared-linux-amd64.deb\n",
            "Reusing existing connection to github.com:443.\n",
            "HTTP request sent, awaiting response... 302 Found\n",
            "Location: https://objects.githubusercontent.com/github-production-release-asset-2e65be/106867604/f6f8a9db-3887-4a1c-9ecb-9781f97395a5?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=releaseassetproduction%2F20250507%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20250507T034229Z&X-Amz-Expires=300&X-Amz-Signature=b0b8bea7c18e37dd1df0da71d9ab482ea736ef25eda01589d4e8bdf5e3bb7449&X-Amz-SignedHeaders=host&response-content-disposition=attachment%3B%20filename%3Dcloudflared-linux-amd64.deb&response-content-type=application%2Foctet-stream [following]\n",
            "--2025-05-07 03:42:29--  https://objects.githubusercontent.com/github-production-release-asset-2e65be/106867604/f6f8a9db-3887-4a1c-9ecb-9781f97395a5?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=releaseassetproduction%2F20250507%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20250507T034229Z&X-Amz-Expires=300&X-Amz-Signature=b0b8bea7c18e37dd1df0da71d9ab482ea736ef25eda01589d4e8bdf5e3bb7449&X-Amz-SignedHeaders=host&response-content-disposition=attachment%3B%20filename%3Dcloudflared-linux-amd64.deb&response-content-type=application%2Foctet-stream\n",
            "Resolving objects.githubusercontent.com (objects.githubusercontent.com)... 185.199.110.133, 185.199.108.133, 185.199.111.133, ...\n",
            "Connecting to objects.githubusercontent.com (objects.githubusercontent.com)|185.199.110.133|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 18546838 (18M) [application/octet-stream]\n",
            "Saving to: ‘/root/cloudflared-linux-amd64.deb’\n",
            "\n",
            "cloudflared-linux-a 100%[===================>]  17.69M  33.9MB/s    in 0.5s    \n",
            "\n",
            "2025-05-07 03:42:30 (33.9 MB/s) - ‘/root/cloudflared-linux-amd64.deb’ saved [18546838/18546838]\n",
            "\n",
            "Selecting previously unselected package cloudflared.\n",
            "(Reading database ... 126102 files and directories currently installed.)\n",
            "Preparing to unpack .../cloudflared-linux-amd64.deb ...\n",
            "Unpacking cloudflared (2025.4.2) ...\n",
            "Setting up cloudflared (2025.4.2) ...\n",
            "Processing triggers for man-db (2.10.2-1) ...\n",
            "[START] Security scan\n",
            "[DONE] Security scan\n",
            "## ComfyUI-Manager: installing dependencies done.\n",
            "** ComfyUI startup time: 2025-05-07 03:42:38.552\n",
            "** Platform: Linux\n",
            "** Python version: 3.11.12 (main, Apr  9 2025, 08:55:54) [GCC 11.4.0]\n",
            "** Python executable: /usr/bin/python3\n",
            "** ComfyUI Path: /content/drive/MyDrive/ComfyUI\n",
            "** ComfyUI Base Folder Path: /content/drive/MyDrive/ComfyUI\n",
            "** User directory: /content/drive/MyDrive/ComfyUI/user\n",
            "** ComfyUI-Manager config path: /content/drive/MyDrive/ComfyUI/user/default/ComfyUI-Manager/config.ini\n",
            "** Log path: /content/drive/MyDrive/ComfyUI/user/comfyui.log\n",
            "\n",
            "Prestartup times for custom nodes:\n",
            "   6.6 seconds: /content/drive/MyDrive/ComfyUI/custom_nodes/ComfyUI-Manager\n",
            "\n",
            "Checkpoint files will always be loaded safely.\n",
            "Total VRAM 15095 MB, total RAM 12978 MB\n",
            "pytorch version: 2.6.0+cu124\n",
            "Set vram state to: NORMAL_VRAM\n",
            "Device: cuda:0 Tesla T4 : cudaMallocAsync\n",
            "Using pytorch attention\n",
            "Python version: 3.11.12 (main, Apr  9 2025, 08:55:54) [GCC 11.4.0]\n",
            "ComfyUI version: 0.3.32\n",
            "ComfyUI frontend version: 1.18.9\n",
            "[Prompt Server] web root: /usr/local/lib/python3.11/dist-packages/comfyui_frontend_package/static\n",
            "Traceback (most recent call last):\n",
            "  File \"/content/drive/MyDrive/ComfyUI/nodes.py\", line 2128, in load_custom_node\n",
            "    module_spec.loader.exec_module(module)\n",
            "  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n",
            "  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_extras/nodes_lt.py\", line 10, in <module>\n",
            "    import av\n",
            "ModuleNotFoundError: No module named 'av'\n",
            "\n",
            "Cannot import /content/drive/MyDrive/ComfyUI/comfy_extras/nodes_lt.py module for custom nodes: No module named 'av'\n",
            "Traceback (most recent call last):\n",
            "  File \"/content/drive/MyDrive/ComfyUI/nodes.py\", line 2128, in load_custom_node\n",
            "    module_spec.loader.exec_module(module)\n",
            "  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n",
            "  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_extras/nodes_video.py\", line 4, in <module>\n",
            "    import av\n",
            "ModuleNotFoundError: No module named 'av'\n",
            "\n",
            "Cannot import /content/drive/MyDrive/ComfyUI/comfy_extras/nodes_video.py module for custom nodes: No module named 'av'\n",
            "Traceback (most recent call last):\n",
            "  File \"/content/drive/MyDrive/ComfyUI/nodes.py\", line 2128, in load_custom_node\n",
            "    module_spec.loader.exec_module(module)\n",
            "  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n",
            "  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_ideogram.py\", line 21, in <module>\n",
            "    from comfy_api_nodes.apinode_utils import (\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/apinode_utils.py\", line 5, in <module>\n",
            "    from comfy_api.input_impl import VideoFromFile\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/__init__.py\", line 1, in <module>\n",
            "    from .video_types import VideoFromFile, VideoFromComponents\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/video_types.py\", line 2, in <module>\n",
            "    from av.container import InputContainer\n",
            "ModuleNotFoundError: No module named 'av'\n",
            "\n",
            "Cannot import /content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_ideogram.py module for custom nodes: No module named 'av'\n",
            "Traceback (most recent call last):\n",
            "  File \"/content/drive/MyDrive/ComfyUI/nodes.py\", line 2128, in load_custom_node\n",
            "    module_spec.loader.exec_module(module)\n",
            "  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n",
            "  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_openai.py\", line 22, in <module>\n",
            "    from comfy_api_nodes.apinode_utils import (\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/apinode_utils.py\", line 5, in <module>\n",
            "    from comfy_api.input_impl import VideoFromFile\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/__init__.py\", line 1, in <module>\n",
            "    from .video_types import VideoFromFile, VideoFromComponents\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/video_types.py\", line 2, in <module>\n",
            "    from av.container import InputContainer\n",
            "ModuleNotFoundError: No module named 'av'\n",
            "\n",
            "Cannot import /content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_openai.py module for custom nodes: No module named 'av'\n",
            "Traceback (most recent call last):\n",
            "  File \"/content/drive/MyDrive/ComfyUI/nodes.py\", line 2128, in load_custom_node\n",
            "    module_spec.loader.exec_module(module)\n",
            "  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n",
            "  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_minimax.py\", line 2, in <module>\n",
            "    from comfy_api.input_impl.video_types import VideoFromFile\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/__init__.py\", line 1, in <module>\n",
            "    from .video_types import VideoFromFile, VideoFromComponents\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/video_types.py\", line 2, in <module>\n",
            "    from av.container import InputContainer\n",
            "ModuleNotFoundError: No module named 'av'\n",
            "\n",
            "Cannot import /content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_minimax.py module for custom nodes: No module named 'av'\n",
            "Traceback (most recent call last):\n",
            "  File \"/content/drive/MyDrive/ComfyUI/nodes.py\", line 2128, in load_custom_node\n",
            "    module_spec.loader.exec_module(module)\n",
            "  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n",
            "  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_veo2.py\", line 8, in <module>\n",
            "    from comfy_api.input_impl.video_types import VideoFromFile\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/__init__.py\", line 1, in <module>\n",
            "    from .video_types import VideoFromFile, VideoFromComponents\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/video_types.py\", line 2, in <module>\n",
            "    from av.container import InputContainer\n",
            "ModuleNotFoundError: No module named 'av'\n",
            "\n",
            "Cannot import /content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_veo2.py module for custom nodes: No module named 'av'\n",
            "Traceback (most recent call last):\n",
            "  File \"/content/drive/MyDrive/ComfyUI/nodes.py\", line 2128, in load_custom_node\n",
            "    module_spec.loader.exec_module(module)\n",
            "  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n",
            "  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_kling.py\", line 59, in <module>\n",
            "    from comfy_api_nodes.apinode_utils import (\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/apinode_utils.py\", line 5, in <module>\n",
            "    from comfy_api.input_impl import VideoFromFile\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/__init__.py\", line 1, in <module>\n",
            "    from .video_types import VideoFromFile, VideoFromComponents\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/video_types.py\", line 2, in <module>\n",
            "    from av.container import InputContainer\n",
            "ModuleNotFoundError: No module named 'av'\n",
            "\n",
            "Cannot import /content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_kling.py module for custom nodes: No module named 'av'\n",
            "Traceback (most recent call last):\n",
            "  File \"/content/drive/MyDrive/ComfyUI/nodes.py\", line 2128, in load_custom_node\n",
            "    module_spec.loader.exec_module(module)\n",
            "  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n",
            "  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_bfl.py\", line 19, in <module>\n",
            "    from comfy_api_nodes.apinode_utils import (\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/apinode_utils.py\", line 5, in <module>\n",
            "    from comfy_api.input_impl import VideoFromFile\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/__init__.py\", line 1, in <module>\n",
            "    from .video_types import VideoFromFile, VideoFromComponents\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/video_types.py\", line 2, in <module>\n",
            "    from av.container import InputContainer\n",
            "ModuleNotFoundError: No module named 'av'\n",
            "\n",
            "Cannot import /content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_bfl.py module for custom nodes: No module named 'av'\n",
            "Traceback (most recent call last):\n",
            "  File \"/content/drive/MyDrive/ComfyUI/nodes.py\", line 2128, in load_custom_node\n",
            "    module_spec.loader.exec_module(module)\n",
            "  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n",
            "  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_luma.py\", line 3, in <module>\n",
            "    from comfy_api.input_impl.video_types import VideoFromFile\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/__init__.py\", line 1, in <module>\n",
            "    from .video_types import VideoFromFile, VideoFromComponents\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/video_types.py\", line 2, in <module>\n",
            "    from av.container import InputContainer\n",
            "ModuleNotFoundError: No module named 'av'\n",
            "\n",
            "Cannot import /content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_luma.py module for custom nodes: No module named 'av'\n",
            "Traceback (most recent call last):\n",
            "  File \"/content/drive/MyDrive/ComfyUI/nodes.py\", line 2128, in load_custom_node\n",
            "    module_spec.loader.exec_module(module)\n",
            "  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n",
            "  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_recraft.py\", line 24, in <module>\n",
            "    from comfy_api_nodes.apinode_utils import (\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/apinode_utils.py\", line 5, in <module>\n",
            "    from comfy_api.input_impl import VideoFromFile\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/__init__.py\", line 1, in <module>\n",
            "    from .video_types import VideoFromFile, VideoFromComponents\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/video_types.py\", line 2, in <module>\n",
            "    from av.container import InputContainer\n",
            "ModuleNotFoundError: No module named 'av'\n",
            "\n",
            "Cannot import /content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_recraft.py module for custom nodes: No module named 'av'\n",
            "Traceback (most recent call last):\n",
            "  File \"/content/drive/MyDrive/ComfyUI/nodes.py\", line 2128, in load_custom_node\n",
            "    module_spec.loader.exec_module(module)\n",
            "  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n",
            "  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_pixverse.py\", line 25, in <module>\n",
            "    from comfy_api_nodes.apinode_utils import (\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/apinode_utils.py\", line 5, in <module>\n",
            "    from comfy_api.input_impl import VideoFromFile\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/__init__.py\", line 1, in <module>\n",
            "    from .video_types import VideoFromFile, VideoFromComponents\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/video_types.py\", line 2, in <module>\n",
            "    from av.container import InputContainer\n",
            "ModuleNotFoundError: No module named 'av'\n",
            "\n",
            "Cannot import /content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_pixverse.py module for custom nodes: No module named 'av'\n",
            "Traceback (most recent call last):\n",
            "  File \"/content/drive/MyDrive/ComfyUI/nodes.py\", line 2128, in load_custom_node\n",
            "    module_spec.loader.exec_module(module)\n",
            "  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n",
            "  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_stability.py\", line 23, in <module>\n",
            "    from comfy_api_nodes.apinode_utils import (\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/apinode_utils.py\", line 5, in <module>\n",
            "    from comfy_api.input_impl import VideoFromFile\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/__init__.py\", line 1, in <module>\n",
            "    from .video_types import VideoFromFile, VideoFromComponents\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/video_types.py\", line 2, in <module>\n",
            "    from av.container import InputContainer\n",
            "ModuleNotFoundError: No module named 'av'\n",
            "\n",
            "Cannot import /content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_stability.py module for custom nodes: No module named 'av'\n",
            "Traceback (most recent call last):\n",
            "  File \"/content/drive/MyDrive/ComfyUI/nodes.py\", line 2128, in load_custom_node\n",
            "    module_spec.loader.exec_module(module)\n",
            "  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n",
            "  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_pika.py\", line 34, in <module>\n",
            "    from comfy_api_nodes.apinode_utils import (\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/apinode_utils.py\", line 5, in <module>\n",
            "    from comfy_api.input_impl import VideoFromFile\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/__init__.py\", line 1, in <module>\n",
            "    from .video_types import VideoFromFile, VideoFromComponents\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/video_types.py\", line 2, in <module>\n",
            "    from av.container import InputContainer\n",
            "ModuleNotFoundError: No module named 'av'\n",
            "\n",
            "Cannot import /content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_pika.py module for custom nodes: No module named 'av'\n",
            "### Loading: ComfyUI-Manager (V3.31.13)\n",
            "[ComfyUI-Manager] network_mode: public\n",
            "### ComfyUI Version: v0.3.32-2-g271c9c5b | Released on '2025-05-06'\n",
            "\n",
            "Import times for custom nodes:\n",
            "   0.5 seconds: /content/drive/MyDrive/ComfyUI/custom_nodes/websocket_image_save.py\n",
            "   0.7 seconds: /content/drive/MyDrive/ComfyUI/custom_nodes/ComfyUI-Manager\n",
            "\n",
            "WARNING: some comfy_api_nodes/ nodes did not import correctly. This may be because they are missing some dependencies.\n",
            "\n",
            "IMPORT FAILED: nodes_ideogram.py\n",
            "IMPORT FAILED: nodes_openai.py\n",
            "IMPORT FAILED: nodes_minimax.py\n",
            "IMPORT FAILED: nodes_veo2.py\n",
            "IMPORT FAILED: nodes_kling.py\n",
            "IMPORT FAILED: nodes_bfl.py\n",
            "IMPORT FAILED: nodes_luma.py\n",
            "IMPORT FAILED: nodes_recraft.py\n",
            "IMPORT FAILED: nodes_pixverse.py\n",
            "IMPORT FAILED: nodes_stability.py\n",
            "IMPORT FAILED: nodes_pika.py\n",
            "\n",
            "This issue might be caused by new missing dependencies added the last time you updated ComfyUI.\n",
            "Please do a: pip install -r requirements.txt\n",
            "\n",
            "WARNING: some comfy_extras/ nodes did not import correctly. This may be because they are missing some dependencies.\n",
            "\n",
            "IMPORT FAILED: nodes_lt.py\n",
            "IMPORT FAILED: nodes_video.py\n",
            "\n",
            "This issue might be caused by new missing dependencies added the last time you updated ComfyUI.\n",
            "Please do a: pip install -r requirements.txt\n",
            "\n",
            "********** ERROR ***********\n",
            "\n",
            "comfyui-workflow-templates is not installed.\n",
            "\n",
            "Please install the updated requirements.txt file by running:\n",
            "/usr/bin/python3 -m pip install -r /content/drive/MyDrive/ComfyUI/requirements.txt\n",
            "\n",
            "This error is happening because the ComfyUI frontend is no longer shipped as part of the main repo but as a pip package instead.\n",
            "\n",
            "If you are on the portable package you can run: update\\update_comfyui.bat to solve this problem\n",
            "\n",
            "********** ERROR ***********\n",
            "[ComfyUI-Manager] default cache updated: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/model-list.json\n",
            "[ComfyUI-Manager] default cache updated: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/custom-node-list.json\n",
            "[ComfyUI-Manager] default cache updated: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/alter-list.json\n",
            "[ComfyUI-Manager] default cache updated: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/github-stats.json\n",
            "[ComfyUI-Manager] default cache updated: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/extension-node-map.json\n",
            "\n",
            "ComfyUI finished loading, trying to launch cloudflared (if it gets stuck here cloudflared is having issues)\n",
            "\n",
            "FETCH ComfyRegistry Data: 5/84\n",
            "This is the URL to access ComfyUI: https://hood-delaware-exactly-editors.trycloudflare.com                                   |\n",
            "FETCH ComfyRegistry Data: 10/84\n",
            "FETCH ComfyRegistry Data: 15/84\n",
            "FETCH ComfyRegistry Data: 20/84\n",
            "FETCH ComfyRegistry Data: 25/84\n",
            "FETCH ComfyRegistry Data: 30/84\n",
            "FETCH ComfyRegistry Data: 35/84\n",
            "FETCH ComfyRegistry Data: 40/84\n",
            "FETCH ComfyRegistry Data: 45/84\n",
            "FETCH ComfyRegistry Data: 50/84\n",
            "FETCH ComfyRegistry Data: 55/84\n",
            "FETCH ComfyRegistry Data: 60/84\n",
            "FETCH ComfyRegistry Data: 65/84\n",
            "FETCH ComfyRegistry Data: 70/84\n",
            "FETCH ComfyRegistry Data: 75/84\n",
            "FETCH ComfyRegistry Data: 80/84\n",
            "FETCH ComfyRegistry Data [DONE]\n",
            "[ComfyUI-Manager] default cache updated: https://api.comfy.org/nodes\n",
            "FETCH DATA from: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/custom-node-list.json [DONE]\n",
            "[ComfyUI-Manager] All startup tasks have been completed.\n",
            "[ComfyUI-Manager] The ComfyRegistry cache update is still in progress, so an outdated cache is being used.\n",
            "FETCH DATA from: /content/drive/MyDrive/ComfyUI/user/default/ComfyUI-Manager/cache/1514988643_custom-node-list.json [DONE]\n",
            "FETCH DATA from: /content/drive/MyDrive/ComfyUI/user/default/ComfyUI-Manager/cache/746607195_github-stats.json [DONE]\n",
            "FETCH DATA from: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/extras.json [DONE]\n",
            "FETCH DATA from: /content/drive/MyDrive/ComfyUI/user/default/ComfyUI-Manager/cache/1742899825_extension-node-map.json [DONE]\n",
            "FETCH DATA from: /content/drive/MyDrive/ComfyUI/user/default/ComfyUI-Manager/cache/1742899825_extension-node-map.json [DONE]\n",
            "100% 12.5M/12.5M [00:00<00:00, 14.7MB/s]\n",
            "Extracted zip file to /content/drive/MyDrive/ComfyUI/custom_nodes/comfyui-jakeupgrade\n",
            "Install: pip packages\n",
            "\n",
            "## ComfyUI-Manager: EXECUTE => ['/usr/bin/python3', '-m', 'pip', 'install', \n",
            "'opencv-python']\n",
            "  \n",
            "## ComfyUI-Manager: EXECUTE => ['/usr/bin/python3', '-m', 'pip', 'install', \n",
            "'piexif']\n",
            " Collecting piexif\n",
            "   Downloading piexif-1.1.3-py2.py3-none-any.whl.metadata (3.7 kB)\n",
            " Downloading piexif-1.1.3-py2.py3-none-any.whl (20 kB)\n",
            " Installing collected packages: piexif\n",
            " Successfully installed piexif-1.1.3\n",
            "[ComfyUI-Manager] skip black listed pip installation: 'torch'\n",
            "\n",
            "## ComfyUI-Manager: EXECUTE => ['/usr/bin/python3', '-m', 'pip', 'install', \n",
            "'numpy<2']\n",
            " \n",
            "## ComfyUI-Manager: EXECUTE => ['/usr/bin/python3', '-m', 'pip', 'install', \n",
            "'numexpr']\n",
            "  \n",
            "## ComfyUI-Manager: EXECUTE => ['/usr/bin/python3', '-m', 'pip', 'install', \n",
            "'simpleeval']\n",
            " Collecting simpleeval\n",
            "   Downloading simpleeval-1.0.3-py3-none-any.whl.metadata (17 kB)\n",
            " Downloading simpleeval-1.0.3-py3-none-any.whl (15 kB)\n",
            " Installing collected packages: simpleeval\n",
            " Successfully installed simpleeval-1.0.3\n",
            "\n",
            "[ComfyUI-Manager] Queued works are completed.\n",
            "{'install': 1}\n",
            "\n",
            "After restarting ComfyUI, please refresh the browser.\n",
            "\n",
            "Restarting... [Legacy Mode]\n",
            "\n",
            "\n",
            "Command: ['/usr/bin/python3', 'main.py', '--dont-print-server']\n",
            "[START] Security scan\n",
            "[DONE] Security scan\n",
            "## ComfyUI-Manager: installing dependencies done.\n",
            "** ComfyUI startup time: 2025-05-07 03:55:57.554\n",
            "** Platform: Linux\n",
            "** Python version: 3.11.12 (main, Apr  9 2025, 08:55:54) [GCC 11.4.0]\n",
            "** Python executable: /usr/bin/python3\n",
            "** ComfyUI Path: /content/drive/MyDrive/ComfyUI\n",
            "** ComfyUI Base Folder Path: /content/drive/MyDrive/ComfyUI\n",
            "** User directory: /content/drive/MyDrive/ComfyUI/user\n",
            "** ComfyUI-Manager config path: /content/drive/MyDrive/ComfyUI/user/default/ComfyUI-Manager/config.ini\n",
            "** Log path: /content/drive/MyDrive/ComfyUI/user/comfyui.log\n",
            "\n",
            "Prestartup times for custom nodes:\n",
            "   5.1 seconds: /content/drive/MyDrive/ComfyUI/custom_nodes/ComfyUI-Manager\n",
            "\n",
            "Checkpoint files will always be loaded safely.\n",
            "Total VRAM 15095 MB, total RAM 12978 MB\n",
            "pytorch version: 2.6.0+cu124\n",
            "Set vram state to: NORMAL_VRAM\n",
            "Device: cuda:0 Tesla T4 : cudaMallocAsync\n",
            "Using pytorch attention\n",
            "Python version: 3.11.12 (main, Apr  9 2025, 08:55:54) [GCC 11.4.0]\n",
            "ComfyUI version: 0.3.32\n",
            "ComfyUI frontend version: 1.18.9\n",
            "[Prompt Server] web root: /usr/local/lib/python3.11/dist-packages/comfyui_frontend_package/static\n",
            "Traceback (most recent call last):\n",
            "  File \"/content/drive/MyDrive/ComfyUI/nodes.py\", line 2128, in load_custom_node\n",
            "    module_spec.loader.exec_module(module)\n",
            "  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n",
            "  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_extras/nodes_lt.py\", line 10, in <module>\n",
            "    import av\n",
            "ModuleNotFoundError: No module named 'av'\n",
            "\n",
            "Cannot import /content/drive/MyDrive/ComfyUI/comfy_extras/nodes_lt.py module for custom nodes: No module named 'av'\n",
            "Traceback (most recent call last):\n",
            "  File \"/content/drive/MyDrive/ComfyUI/nodes.py\", line 2128, in load_custom_node\n",
            "    module_spec.loader.exec_module(module)\n",
            "  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n",
            "  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_extras/nodes_video.py\", line 4, in <module>\n",
            "    import av\n",
            "ModuleNotFoundError: No module named 'av'\n",
            "\n",
            "Cannot import /content/drive/MyDrive/ComfyUI/comfy_extras/nodes_video.py module for custom nodes: No module named 'av'\n",
            "Traceback (most recent call last):\n",
            "  File \"/content/drive/MyDrive/ComfyUI/nodes.py\", line 2128, in load_custom_node\n",
            "    module_spec.loader.exec_module(module)\n",
            "  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n",
            "  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_ideogram.py\", line 21, in <module>\n",
            "    from comfy_api_nodes.apinode_utils import (\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/apinode_utils.py\", line 5, in <module>\n",
            "    from comfy_api.input_impl import VideoFromFile\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/__init__.py\", line 1, in <module>\n",
            "    from .video_types import VideoFromFile, VideoFromComponents\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/video_types.py\", line 2, in <module>\n",
            "    from av.container import InputContainer\n",
            "ModuleNotFoundError: No module named 'av'\n",
            "\n",
            "Cannot import /content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_ideogram.py module for custom nodes: No module named 'av'\n",
            "Traceback (most recent call last):\n",
            "  File \"/content/drive/MyDrive/ComfyUI/nodes.py\", line 2128, in load_custom_node\n",
            "    module_spec.loader.exec_module(module)\n",
            "  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n",
            "  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_openai.py\", line 22, in <module>\n",
            "    from comfy_api_nodes.apinode_utils import (\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/apinode_utils.py\", line 5, in <module>\n",
            "    from comfy_api.input_impl import VideoFromFile\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/__init__.py\", line 1, in <module>\n",
            "    from .video_types import VideoFromFile, VideoFromComponents\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/video_types.py\", line 2, in <module>\n",
            "    from av.container import InputContainer\n",
            "ModuleNotFoundError: No module named 'av'\n",
            "\n",
            "Cannot import /content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_openai.py module for custom nodes: No module named 'av'\n",
            "Traceback (most recent call last):\n",
            "  File \"/content/drive/MyDrive/ComfyUI/nodes.py\", line 2128, in load_custom_node\n",
            "    module_spec.loader.exec_module(module)\n",
            "  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n",
            "  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_minimax.py\", line 2, in <module>\n",
            "    from comfy_api.input_impl.video_types import VideoFromFile\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/__init__.py\", line 1, in <module>\n",
            "    from .video_types import VideoFromFile, VideoFromComponents\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/video_types.py\", line 2, in <module>\n",
            "    from av.container import InputContainer\n",
            "ModuleNotFoundError: No module named 'av'\n",
            "\n",
            "Cannot import /content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_minimax.py module for custom nodes: No module named 'av'\n",
            "Traceback (most recent call last):\n",
            "  File \"/content/drive/MyDrive/ComfyUI/nodes.py\", line 2128, in load_custom_node\n",
            "    module_spec.loader.exec_module(module)\n",
            "  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n",
            "  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_veo2.py\", line 8, in <module>\n",
            "    from comfy_api.input_impl.video_types import VideoFromFile\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/__init__.py\", line 1, in <module>\n",
            "    from .video_types import VideoFromFile, VideoFromComponents\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/video_types.py\", line 2, in <module>\n",
            "    from av.container import InputContainer\n",
            "ModuleNotFoundError: No module named 'av'\n",
            "\n",
            "Cannot import /content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_veo2.py module for custom nodes: No module named 'av'\n",
            "Traceback (most recent call last):\n",
            "  File \"/content/drive/MyDrive/ComfyUI/nodes.py\", line 2128, in load_custom_node\n",
            "    module_spec.loader.exec_module(module)\n",
            "  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n",
            "  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_kling.py\", line 59, in <module>\n",
            "    from comfy_api_nodes.apinode_utils import (\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/apinode_utils.py\", line 5, in <module>\n",
            "    from comfy_api.input_impl import VideoFromFile\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/__init__.py\", line 1, in <module>\n",
            "    from .video_types import VideoFromFile, VideoFromComponents\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/video_types.py\", line 2, in <module>\n",
            "    from av.container import InputContainer\n",
            "ModuleNotFoundError: No module named 'av'\n",
            "\n",
            "Cannot import /content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_kling.py module for custom nodes: No module named 'av'\n",
            "Traceback (most recent call last):\n",
            "  File \"/content/drive/MyDrive/ComfyUI/nodes.py\", line 2128, in load_custom_node\n",
            "    module_spec.loader.exec_module(module)\n",
            "  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n",
            "  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_bfl.py\", line 19, in <module>\n",
            "    from comfy_api_nodes.apinode_utils import (\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/apinode_utils.py\", line 5, in <module>\n",
            "    from comfy_api.input_impl import VideoFromFile\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/__init__.py\", line 1, in <module>\n",
            "    from .video_types import VideoFromFile, VideoFromComponents\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/video_types.py\", line 2, in <module>\n",
            "    from av.container import InputContainer\n",
            "ModuleNotFoundError: No module named 'av'\n",
            "\n",
            "Cannot import /content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_bfl.py module for custom nodes: No module named 'av'\n",
            "Traceback (most recent call last):\n",
            "  File \"/content/drive/MyDrive/ComfyUI/nodes.py\", line 2128, in load_custom_node\n",
            "    module_spec.loader.exec_module(module)\n",
            "  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n",
            "  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_luma.py\", line 3, in <module>\n",
            "    from comfy_api.input_impl.video_types import VideoFromFile\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/__init__.py\", line 1, in <module>\n",
            "    from .video_types import VideoFromFile, VideoFromComponents\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/video_types.py\", line 2, in <module>\n",
            "    from av.container import InputContainer\n",
            "ModuleNotFoundError: No module named 'av'\n",
            "\n",
            "Cannot import /content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_luma.py module for custom nodes: No module named 'av'\n",
            "Traceback (most recent call last):\n",
            "  File \"/content/drive/MyDrive/ComfyUI/nodes.py\", line 2128, in load_custom_node\n",
            "    module_spec.loader.exec_module(module)\n",
            "  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n",
            "  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_recraft.py\", line 24, in <module>\n",
            "    from comfy_api_nodes.apinode_utils import (\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/apinode_utils.py\", line 5, in <module>\n",
            "    from comfy_api.input_impl import VideoFromFile\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/__init__.py\", line 1, in <module>\n",
            "    from .video_types import VideoFromFile, VideoFromComponents\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/video_types.py\", line 2, in <module>\n",
            "    from av.container import InputContainer\n",
            "ModuleNotFoundError: No module named 'av'\n",
            "\n",
            "Cannot import /content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_recraft.py module for custom nodes: No module named 'av'\n",
            "Traceback (most recent call last):\n",
            "  File \"/content/drive/MyDrive/ComfyUI/nodes.py\", line 2128, in load_custom_node\n",
            "    module_spec.loader.exec_module(module)\n",
            "  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n",
            "  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_pixverse.py\", line 25, in <module>\n",
            "    from comfy_api_nodes.apinode_utils import (\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/apinode_utils.py\", line 5, in <module>\n",
            "    from comfy_api.input_impl import VideoFromFile\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/__init__.py\", line 1, in <module>\n",
            "    from .video_types import VideoFromFile, VideoFromComponents\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/video_types.py\", line 2, in <module>\n",
            "    from av.container import InputContainer\n",
            "ModuleNotFoundError: No module named 'av'\n",
            "\n",
            "Cannot import /content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_pixverse.py module for custom nodes: No module named 'av'\n",
            "Traceback (most recent call last):\n",
            "  File \"/content/drive/MyDrive/ComfyUI/nodes.py\", line 2128, in load_custom_node\n",
            "    module_spec.loader.exec_module(module)\n",
            "  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n",
            "  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_stability.py\", line 23, in <module>\n",
            "    from comfy_api_nodes.apinode_utils import (\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/apinode_utils.py\", line 5, in <module>\n",
            "    from comfy_api.input_impl import VideoFromFile\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/__init__.py\", line 1, in <module>\n",
            "    from .video_types import VideoFromFile, VideoFromComponents\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/video_types.py\", line 2, in <module>\n",
            "    from av.container import InputContainer\n",
            "ModuleNotFoundError: No module named 'av'\n",
            "\n",
            "Cannot import /content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_stability.py module for custom nodes: No module named 'av'\n",
            "Traceback (most recent call last):\n",
            "  File \"/content/drive/MyDrive/ComfyUI/nodes.py\", line 2128, in load_custom_node\n",
            "    module_spec.loader.exec_module(module)\n",
            "  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n",
            "  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_pika.py\", line 34, in <module>\n",
            "    from comfy_api_nodes.apinode_utils import (\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/apinode_utils.py\", line 5, in <module>\n",
            "    from comfy_api.input_impl import VideoFromFile\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/__init__.py\", line 1, in <module>\n",
            "    from .video_types import VideoFromFile, VideoFromComponents\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/video_types.py\", line 2, in <module>\n",
            "    from av.container import InputContainer\n",
            "ModuleNotFoundError: No module named 'av'\n",
            "\n",
            "Cannot import /content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_pika.py module for custom nodes: No module named 'av'\n",
            "### Loading: ComfyUI-Manager (V3.31.13)\n",
            "[ComfyUI-Manager] network_mode: public\n",
            "### ComfyUI Version: v0.3.32-2-g271c9c5b | Released on '2025-05-06'\n",
            "[ComfyUI-Manager] default cache updated: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/custom-node-list.json\n",
            "[ComfyUI-Manager] default cache updated: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/model-list.json\n",
            "[ComfyUI-Manager] default cache updated: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/alter-list.json\n",
            "[ComfyUI-Manager] default cache updated: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/extension-node-map.json\n",
            "[ComfyUI-Manager] default cache updated: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/github-stats.json\n",
            "----------Jake Upgrade Nodes Loaded----------\n",
            "\n",
            "Import times for custom nodes:\n",
            "   0.0 seconds: /content/drive/MyDrive/ComfyUI/custom_nodes/websocket_image_save.py\n",
            "   1.1 seconds: /content/drive/MyDrive/ComfyUI/custom_nodes/ComfyUI-Manager\n",
            "   1.3 seconds: /content/drive/MyDrive/ComfyUI/custom_nodes/comfyui-jakeupgrade\n",
            "\n",
            "WARNING: some comfy_api_nodes/ nodes did not import correctly. This may be because they are missing some dependencies.\n",
            "\n",
            "IMPORT FAILED: nodes_ideogram.py\n",
            "IMPORT FAILED: nodes_openai.py\n",
            "IMPORT FAILED: nodes_minimax.py\n",
            "IMPORT FAILED: nodes_veo2.py\n",
            "IMPORT FAILED: nodes_kling.py\n",
            "IMPORT FAILED: nodes_bfl.py\n",
            "IMPORT FAILED: nodes_luma.py\n",
            "IMPORT FAILED: nodes_recraft.py\n",
            "IMPORT FAILED: nodes_pixverse.py\n",
            "IMPORT FAILED: nodes_stability.py\n",
            "IMPORT FAILED: nodes_pika.py\n",
            "\n",
            "This issue might be caused by new missing dependencies added the last time you updated ComfyUI.\n",
            "Please do a: pip install -r requirements.txt\n",
            "\n",
            "WARNING: some comfy_extras/ nodes did not import correctly. This may be because they are missing some dependencies.\n",
            "\n",
            "IMPORT FAILED: nodes_lt.py\n",
            "IMPORT FAILED: nodes_video.py\n",
            "\n",
            "This issue might be caused by new missing dependencies added the last time you updated ComfyUI.\n",
            "Please do a: pip install -r requirements.txt\n",
            "\n",
            "********** ERROR ***********\n",
            "\n",
            "comfyui-workflow-templates is not installed.\n",
            "\n",
            "Please install the updated requirements.txt file by running:\n",
            "/usr/bin/python3 -m pip install -r /content/drive/MyDrive/ComfyUI/requirements.txt\n",
            "\n",
            "This error is happening because the ComfyUI frontend is no longer shipped as part of the main repo but as a pip package instead.\n",
            "\n",
            "If you are on the portable package you can run: update\\update_comfyui.bat to solve this problem\n",
            "\n",
            "********** ERROR ***********\n",
            "FETCH ComfyRegistry Data: 5/84\n",
            "FETCH ComfyRegistry Data: 10/84\n",
            "FETCH ComfyRegistry Data: 15/84\n",
            "FETCH ComfyRegistry Data: 20/84\n",
            "FETCH ComfyRegistry Data: 25/84\n",
            "FETCH ComfyRegistry Data: 30/84\n",
            "FETCH ComfyRegistry Data: 35/84\n",
            "FETCH ComfyRegistry Data: 40/84\n",
            "FETCH ComfyRegistry Data: 45/84\n",
            "FETCH ComfyRegistry Data: 50/84\n",
            "FETCH ComfyRegistry Data: 55/84\n",
            "FETCH ComfyRegistry Data: 60/84\n",
            "FETCH ComfyRegistry Data: 65/84\n",
            "FETCH ComfyRegistry Data: 70/84\n",
            "FETCH ComfyRegistry Data: 75/84\n",
            "FETCH ComfyRegistry Data: 80/84\n",
            "FETCH ComfyRegistry Data [DONE]\n",
            "[ComfyUI-Manager] default cache updated: https://api.comfy.org/nodes\n",
            "FETCH DATA from: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/custom-node-list.json [DONE]\n",
            "[ComfyUI-Manager] All startup tasks have been completed.\n",
            "[ComfyUI-Manager] The ComfyRegistry cache update is still in progress, so an outdated cache is being used.\n",
            "FETCH DATA from: /content/drive/MyDrive/ComfyUI/user/default/ComfyUI-Manager/cache/1514988643_custom-node-list.json [DONE]\n",
            "FETCH DATA from: /content/drive/MyDrive/ComfyUI/user/default/ComfyUI-Manager/cache/746607195_github-stats.json [DONE]\n",
            "FETCH DATA from: /content/drive/MyDrive/ComfyUI/user/default/ComfyUI-Manager/cache/832903789_extras.json [DONE]\n",
            "FETCH DATA from: /content/drive/MyDrive/ComfyUI/user/default/ComfyUI-Manager/cache/1742899825_extension-node-map.json [DONE]\n",
            "FETCH DATA from: /content/drive/MyDrive/ComfyUI/user/default/ComfyUI-Manager/cache/1742899825_extension-node-map.json [DONE]\n",
            "Updating: comfyui-jakeupgrade\n",
            "[ComfyUI-Manager] Queued works are completed.\n",
            "{'update': 1}\n",
            "\n",
            "After restarting ComfyUI, please refresh the browser.\n",
            "\n",
            "Restarting... [Legacy Mode]\n",
            "\n",
            "\n",
            "Command: ['/usr/bin/python3', 'main.py', '--dont-print-server']\n",
            "[START] Security scan\n",
            "[DONE] Security scan\n",
            "## ComfyUI-Manager: installing dependencies done.\n",
            "** ComfyUI startup time: 2025-05-07 03:59:02.445\n",
            "** Platform: Linux\n",
            "** Python version: 3.11.12 (main, Apr  9 2025, 08:55:54) [GCC 11.4.0]\n",
            "** Python executable: /usr/bin/python3\n",
            "** ComfyUI Path: /content/drive/MyDrive/ComfyUI\n",
            "** ComfyUI Base Folder Path: /content/drive/MyDrive/ComfyUI\n",
            "** User directory: /content/drive/MyDrive/ComfyUI/user\n",
            "** ComfyUI-Manager config path: /content/drive/MyDrive/ComfyUI/user/default/ComfyUI-Manager/config.ini\n",
            "** Log path: /content/drive/MyDrive/ComfyUI/user/comfyui.log\n",
            "\n",
            "Prestartup times for custom nodes:\n",
            "   5.5 seconds: /content/drive/MyDrive/ComfyUI/custom_nodes/ComfyUI-Manager\n",
            "\n",
            "Checkpoint files will always be loaded safely.\n",
            "Total VRAM 15095 MB, total RAM 12978 MB\n",
            "pytorch version: 2.6.0+cu124\n",
            "Set vram state to: NORMAL_VRAM\n",
            "Device: cuda:0 Tesla T4 : cudaMallocAsync\n",
            "Using pytorch attention\n",
            "Python version: 3.11.12 (main, Apr  9 2025, 08:55:54) [GCC 11.4.0]\n",
            "ComfyUI version: 0.3.32\n",
            "ComfyUI frontend version: 1.18.9\n",
            "[Prompt Server] web root: /usr/local/lib/python3.11/dist-packages/comfyui_frontend_package/static\n",
            "Traceback (most recent call last):\n",
            "  File \"/content/drive/MyDrive/ComfyUI/nodes.py\", line 2128, in load_custom_node\n",
            "    module_spec.loader.exec_module(module)\n",
            "  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n",
            "  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_extras/nodes_lt.py\", line 10, in <module>\n",
            "    import av\n",
            "ModuleNotFoundError: No module named 'av'\n",
            "\n",
            "Cannot import /content/drive/MyDrive/ComfyUI/comfy_extras/nodes_lt.py module for custom nodes: No module named 'av'\n",
            "Traceback (most recent call last):\n",
            "  File \"/content/drive/MyDrive/ComfyUI/nodes.py\", line 2128, in load_custom_node\n",
            "    module_spec.loader.exec_module(module)\n",
            "  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n",
            "  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_extras/nodes_video.py\", line 4, in <module>\n",
            "    import av\n",
            "ModuleNotFoundError: No module named 'av'\n",
            "\n",
            "Cannot import /content/drive/MyDrive/ComfyUI/comfy_extras/nodes_video.py module for custom nodes: No module named 'av'\n",
            "Traceback (most recent call last):\n",
            "  File \"/content/drive/MyDrive/ComfyUI/nodes.py\", line 2128, in load_custom_node\n",
            "    module_spec.loader.exec_module(module)\n",
            "  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n",
            "  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_ideogram.py\", line 21, in <module>\n",
            "    from comfy_api_nodes.apinode_utils import (\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/apinode_utils.py\", line 5, in <module>\n",
            "    from comfy_api.input_impl import VideoFromFile\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/__init__.py\", line 1, in <module>\n",
            "    from .video_types import VideoFromFile, VideoFromComponents\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/video_types.py\", line 2, in <module>\n",
            "    from av.container import InputContainer\n",
            "ModuleNotFoundError: No module named 'av'\n",
            "\n",
            "Cannot import /content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_ideogram.py module for custom nodes: No module named 'av'\n",
            "Traceback (most recent call last):\n",
            "  File \"/content/drive/MyDrive/ComfyUI/nodes.py\", line 2128, in load_custom_node\n",
            "    module_spec.loader.exec_module(module)\n",
            "  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n",
            "  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_openai.py\", line 22, in <module>\n",
            "    from comfy_api_nodes.apinode_utils import (\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/apinode_utils.py\", line 5, in <module>\n",
            "    from comfy_api.input_impl import VideoFromFile\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/__init__.py\", line 1, in <module>\n",
            "    from .video_types import VideoFromFile, VideoFromComponents\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/video_types.py\", line 2, in <module>\n",
            "    from av.container import InputContainer\n",
            "ModuleNotFoundError: No module named 'av'\n",
            "\n",
            "Cannot import /content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_openai.py module for custom nodes: No module named 'av'\n",
            "Traceback (most recent call last):\n",
            "  File \"/content/drive/MyDrive/ComfyUI/nodes.py\", line 2128, in load_custom_node\n",
            "    module_spec.loader.exec_module(module)\n",
            "  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n",
            "  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_minimax.py\", line 2, in <module>\n",
            "    from comfy_api.input_impl.video_types import VideoFromFile\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/__init__.py\", line 1, in <module>\n",
            "    from .video_types import VideoFromFile, VideoFromComponents\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/video_types.py\", line 2, in <module>\n",
            "    from av.container import InputContainer\n",
            "ModuleNotFoundError: No module named 'av'\n",
            "\n",
            "Cannot import /content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_minimax.py module for custom nodes: No module named 'av'\n",
            "Traceback (most recent call last):\n",
            "  File \"/content/drive/MyDrive/ComfyUI/nodes.py\", line 2128, in load_custom_node\n",
            "    module_spec.loader.exec_module(module)\n",
            "  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n",
            "  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_veo2.py\", line 8, in <module>\n",
            "    from comfy_api.input_impl.video_types import VideoFromFile\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/__init__.py\", line 1, in <module>\n",
            "    from .video_types import VideoFromFile, VideoFromComponents\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/video_types.py\", line 2, in <module>\n",
            "    from av.container import InputContainer\n",
            "ModuleNotFoundError: No module named 'av'\n",
            "\n",
            "Cannot import /content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_veo2.py module for custom nodes: No module named 'av'\n",
            "Traceback (most recent call last):\n",
            "  File \"/content/drive/MyDrive/ComfyUI/nodes.py\", line 2128, in load_custom_node\n",
            "    module_spec.loader.exec_module(module)\n",
            "  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n",
            "  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_kling.py\", line 59, in <module>\n",
            "    from comfy_api_nodes.apinode_utils import (\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/apinode_utils.py\", line 5, in <module>\n",
            "    from comfy_api.input_impl import VideoFromFile\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/__init__.py\", line 1, in <module>\n",
            "    from .video_types import VideoFromFile, VideoFromComponents\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/video_types.py\", line 2, in <module>\n",
            "    from av.container import InputContainer\n",
            "ModuleNotFoundError: No module named 'av'\n",
            "\n",
            "Cannot import /content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_kling.py module for custom nodes: No module named 'av'\n",
            "Traceback (most recent call last):\n",
            "  File \"/content/drive/MyDrive/ComfyUI/nodes.py\", line 2128, in load_custom_node\n",
            "    module_spec.loader.exec_module(module)\n",
            "  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n",
            "  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_bfl.py\", line 19, in <module>\n",
            "    from comfy_api_nodes.apinode_utils import (\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/apinode_utils.py\", line 5, in <module>\n",
            "    from comfy_api.input_impl import VideoFromFile\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/__init__.py\", line 1, in <module>\n",
            "    from .video_types import VideoFromFile, VideoFromComponents\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/video_types.py\", line 2, in <module>\n",
            "    from av.container import InputContainer\n",
            "ModuleNotFoundError: No module named 'av'\n",
            "\n",
            "Cannot import /content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_bfl.py module for custom nodes: No module named 'av'\n",
            "Traceback (most recent call last):\n",
            "  File \"/content/drive/MyDrive/ComfyUI/nodes.py\", line 2128, in load_custom_node\n",
            "    module_spec.loader.exec_module(module)\n",
            "  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n",
            "  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_luma.py\", line 3, in <module>\n",
            "    from comfy_api.input_impl.video_types import VideoFromFile\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/__init__.py\", line 1, in <module>\n",
            "    from .video_types import VideoFromFile, VideoFromComponents\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/video_types.py\", line 2, in <module>\n",
            "    from av.container import InputContainer\n",
            "ModuleNotFoundError: No module named 'av'\n",
            "\n",
            "Cannot import /content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_luma.py module for custom nodes: No module named 'av'\n",
            "Traceback (most recent call last):\n",
            "  File \"/content/drive/MyDrive/ComfyUI/nodes.py\", line 2128, in load_custom_node\n",
            "    module_spec.loader.exec_module(module)\n",
            "  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n",
            "  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_recraft.py\", line 24, in <module>\n",
            "    from comfy_api_nodes.apinode_utils import (\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/apinode_utils.py\", line 5, in <module>\n",
            "    from comfy_api.input_impl import VideoFromFile\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/__init__.py\", line 1, in <module>\n",
            "    from .video_types import VideoFromFile, VideoFromComponents\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/video_types.py\", line 2, in <module>\n",
            "    from av.container import InputContainer\n",
            "ModuleNotFoundError: No module named 'av'\n",
            "\n",
            "Cannot import /content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_recraft.py module for custom nodes: No module named 'av'\n",
            "Traceback (most recent call last):\n",
            "  File \"/content/drive/MyDrive/ComfyUI/nodes.py\", line 2128, in load_custom_node\n",
            "    module_spec.loader.exec_module(module)\n",
            "  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n",
            "  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_pixverse.py\", line 25, in <module>\n",
            "    from comfy_api_nodes.apinode_utils import (\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/apinode_utils.py\", line 5, in <module>\n",
            "    from comfy_api.input_impl import VideoFromFile\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/__init__.py\", line 1, in <module>\n",
            "    from .video_types import VideoFromFile, VideoFromComponents\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/video_types.py\", line 2, in <module>\n",
            "    from av.container import InputContainer\n",
            "ModuleNotFoundError: No module named 'av'\n",
            "\n",
            "Cannot import /content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_pixverse.py module for custom nodes: No module named 'av'\n",
            "Traceback (most recent call last):\n",
            "  File \"/content/drive/MyDrive/ComfyUI/nodes.py\", line 2128, in load_custom_node\n",
            "    module_spec.loader.exec_module(module)\n",
            "  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n",
            "  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_stability.py\", line 23, in <module>\n",
            "    from comfy_api_nodes.apinode_utils import (\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/apinode_utils.py\", line 5, in <module>\n",
            "    from comfy_api.input_impl import VideoFromFile\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/__init__.py\", line 1, in <module>\n",
            "    from .video_types import VideoFromFile, VideoFromComponents\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/video_types.py\", line 2, in <module>\n",
            "    from av.container import InputContainer\n",
            "ModuleNotFoundError: No module named 'av'\n",
            "\n",
            "Cannot import /content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_stability.py module for custom nodes: No module named 'av'\n",
            "Traceback (most recent call last):\n",
            "  File \"/content/drive/MyDrive/ComfyUI/nodes.py\", line 2128, in load_custom_node\n",
            "    module_spec.loader.exec_module(module)\n",
            "  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n",
            "  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_pika.py\", line 34, in <module>\n",
            "    from comfy_api_nodes.apinode_utils import (\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/apinode_utils.py\", line 5, in <module>\n",
            "    from comfy_api.input_impl import VideoFromFile\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/__init__.py\", line 1, in <module>\n",
            "    from .video_types import VideoFromFile, VideoFromComponents\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/video_types.py\", line 2, in <module>\n",
            "    from av.container import InputContainer\n",
            "ModuleNotFoundError: No module named 'av'\n",
            "\n",
            "Cannot import /content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_pika.py module for custom nodes: No module named 'av'\n",
            "### Loading: ComfyUI-Manager (V3.31.13)\n",
            "[ComfyUI-Manager] network_mode: public\n",
            "### ComfyUI Version: v0.3.32-2-g271c9c5b | Released on '2025-05-06'\n",
            "[ComfyUI-Manager] default cache updated: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/model-list.json\n",
            "[ComfyUI-Manager] default cache updated: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/github-stats.json\n",
            "[ComfyUI-Manager] default cache updated: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/alter-list.json\n",
            "[ComfyUI-Manager] default cache updated: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/extension-node-map.json\n",
            "[ComfyUI-Manager] default cache updated: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/custom-node-list.json\n",
            "----------Jake Upgrade Nodes Loaded----------\n",
            "\n",
            "Import times for custom nodes:\n",
            "   0.0 seconds: /content/drive/MyDrive/ComfyUI/custom_nodes/websocket_image_save.py\n",
            "   0.5 seconds: /content/drive/MyDrive/ComfyUI/custom_nodes/comfyui-jakeupgrade\n",
            "   0.6 seconds: /content/drive/MyDrive/ComfyUI/custom_nodes/ComfyUI-Manager\n",
            "\n",
            "WARNING: some comfy_api_nodes/ nodes did not import correctly. This may be because they are missing some dependencies.\n",
            "\n",
            "IMPORT FAILED: nodes_ideogram.py\n",
            "IMPORT FAILED: nodes_openai.py\n",
            "IMPORT FAILED: nodes_minimax.py\n",
            "IMPORT FAILED: nodes_veo2.py\n",
            "IMPORT FAILED: nodes_kling.py\n",
            "IMPORT FAILED: nodes_bfl.py\n",
            "IMPORT FAILED: nodes_luma.py\n",
            "IMPORT FAILED: nodes_recraft.py\n",
            "IMPORT FAILED: nodes_pixverse.py\n",
            "IMPORT FAILED: nodes_stability.py\n",
            "IMPORT FAILED: nodes_pika.py\n",
            "\n",
            "This issue might be caused by new missing dependencies added the last time you updated ComfyUI.\n",
            "Please do a: pip install -r requirements.txt\n",
            "\n",
            "WARNING: some comfy_extras/ nodes did not import correctly. This may be because they are missing some dependencies.\n",
            "\n",
            "IMPORT FAILED: nodes_lt.py\n",
            "IMPORT FAILED: nodes_video.py\n",
            "\n",
            "This issue might be caused by new missing dependencies added the last time you updated ComfyUI.\n",
            "Please do a: pip install -r requirements.txt\n",
            "\n",
            "********** ERROR ***********\n",
            "\n",
            "comfyui-workflow-templates is not installed.\n",
            "\n",
            "Please install the updated requirements.txt file by running:\n",
            "/usr/bin/python3 -m pip install -r /content/drive/MyDrive/ComfyUI/requirements.txt\n",
            "\n",
            "This error is happening because the ComfyUI frontend is no longer shipped as part of the main repo but as a pip package instead.\n",
            "\n",
            "If you are on the portable package you can run: update\\update_comfyui.bat to solve this problem\n",
            "\n",
            "********** ERROR ***********\n",
            "FETCH ComfyRegistry Data: 5/84\n",
            "FETCH ComfyRegistry Data: 10/84\n",
            "FETCH ComfyRegistry Data: 15/84\n",
            "[ComfyUI-Manager] The ComfyRegistry cache update is still in progress, so an outdated cache is being used.\n",
            "FETCH DATA from: /content/drive/MyDrive/ComfyUI/user/default/ComfyUI-Manager/cache/1514988643_custom-node-list.json [DONE]\n",
            "FETCH DATA from: /content/drive/MyDrive/ComfyUI/user/default/ComfyUI-Manager/cache/746607195_github-stats.json [DONE]\n",
            "FETCH DATA from: /content/drive/MyDrive/ComfyUI/user/default/ComfyUI-Manager/cache/832903789_extras.json [DONE]\n",
            "FETCH ComfyRegistry Data: 20/84\n",
            "FETCH ComfyRegistry Data: 25/84\n",
            "FETCH ComfyRegistry Data: 30/84\n",
            "FETCH DATA from: /content/drive/MyDrive/ComfyUI/user/default/ComfyUI-Manager/cache/1742899825_extension-node-map.json [DONE]\n",
            "FETCH ComfyRegistry Data: 35/84\n",
            "FETCH DATA from: /content/drive/MyDrive/ComfyUI/user/default/ComfyUI-Manager/cache/1742899825_extension-node-map.json [DONE]\n",
            "FETCH ComfyRegistry Data: 40/84\n",
            "FETCH ComfyRegistry Data: 45/84\n",
            "FETCH ComfyRegistry Data: 50/84\n",
            "Updating: comfyui-jakeupgrade\n",
            "[ComfyUI-Manager] Queued works are completed.\n",
            "{'update': 1}\n",
            "\n",
            "After restarting ComfyUI, please refresh the browser.\n",
            "FETCH ComfyRegistry Data: 55/84\n",
            "FETCH ComfyRegistry Data: 60/84\n",
            "FETCH ComfyRegistry Data: 65/84\n",
            "FETCH ComfyRegistry Data: 70/84\n",
            "FETCH ComfyRegistry Data: 75/84\n",
            "FETCH ComfyRegistry Data: 80/84\n",
            "FETCH ComfyRegistry Data [DONE]\n",
            "[ComfyUI-Manager] default cache updated: https://api.comfy.org/nodes\n",
            "FETCH DATA from: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/custom-node-list.json [DONE]\n",
            "[ComfyUI-Manager] All startup tasks have been completed.\n",
            "[ComfyUI-Manager] The ComfyRegistry cache update is still in progress, so an outdated cache is being used.\n",
            "FETCH DATA from: /content/drive/MyDrive/ComfyUI/user/default/ComfyUI-Manager/cache/1514988643_custom-node-list.json [DONE]\n",
            "FETCH DATA from: /content/drive/MyDrive/ComfyUI/user/default/ComfyUI-Manager/cache/746607195_github-stats.json [DONE]\n",
            "FETCH DATA from: /content/drive/MyDrive/ComfyUI/user/default/ComfyUI-Manager/cache/832903789_extras.json [DONE]\n",
            "FETCH DATA from: /content/drive/MyDrive/ComfyUI/user/default/ComfyUI-Manager/cache/1742899825_extension-node-map.json [DONE]\n",
            "FETCH DATA from: /content/drive/MyDrive/ComfyUI/user/default/ComfyUI-Manager/cache/1742899825_extension-node-map.json [DONE]\n",
            "Updating: comfyui-jakeupgrade\n",
            "[ComfyUI-Manager] Queued works are completed.\n",
            "{'update': 1}\n",
            "\n",
            "After restarting ComfyUI, please refresh the browser.\n",
            "\n",
            "Restarting... [Legacy Mode]\n",
            "\n",
            "\n",
            "Command: ['/usr/bin/python3', 'main.py', '--dont-print-server']\n",
            "[START] Security scan\n",
            "[DONE] Security scan\n",
            "## ComfyUI-Manager: installing dependencies done.\n",
            "** ComfyUI startup time: 2025-05-07 04:01:46.053\n",
            "** Platform: Linux\n",
            "** Python version: 3.11.12 (main, Apr  9 2025, 08:55:54) [GCC 11.4.0]\n",
            "** Python executable: /usr/bin/python3\n",
            "** ComfyUI Path: /content/drive/MyDrive/ComfyUI\n",
            "** ComfyUI Base Folder Path: /content/drive/MyDrive/ComfyUI\n",
            "** User directory: /content/drive/MyDrive/ComfyUI/user\n",
            "** ComfyUI-Manager config path: /content/drive/MyDrive/ComfyUI/user/default/ComfyUI-Manager/config.ini\n",
            "** Log path: /content/drive/MyDrive/ComfyUI/user/comfyui.log\n",
            "\n",
            "Prestartup times for custom nodes:\n",
            "   4.9 seconds: /content/drive/MyDrive/ComfyUI/custom_nodes/ComfyUI-Manager\n",
            "\n",
            "Checkpoint files will always be loaded safely.\n",
            "Total VRAM 15095 MB, total RAM 12978 MB\n",
            "pytorch version: 2.6.0+cu124\n",
            "Set vram state to: NORMAL_VRAM\n",
            "Device: cuda:0 Tesla T4 : cudaMallocAsync\n",
            "Using pytorch attention\n",
            "Python version: 3.11.12 (main, Apr  9 2025, 08:55:54) [GCC 11.4.0]\n",
            "ComfyUI version: 0.3.32\n",
            "ComfyUI frontend version: 1.18.9\n",
            "[Prompt Server] web root: /usr/local/lib/python3.11/dist-packages/comfyui_frontend_package/static\n",
            "Traceback (most recent call last):\n",
            "  File \"/content/drive/MyDrive/ComfyUI/nodes.py\", line 2128, in load_custom_node\n",
            "    module_spec.loader.exec_module(module)\n",
            "  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n",
            "  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_extras/nodes_lt.py\", line 10, in <module>\n",
            "    import av\n",
            "ModuleNotFoundError: No module named 'av'\n",
            "\n",
            "Cannot import /content/drive/MyDrive/ComfyUI/comfy_extras/nodes_lt.py module for custom nodes: No module named 'av'\n",
            "Traceback (most recent call last):\n",
            "  File \"/content/drive/MyDrive/ComfyUI/nodes.py\", line 2128, in load_custom_node\n",
            "    module_spec.loader.exec_module(module)\n",
            "  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n",
            "  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_extras/nodes_video.py\", line 4, in <module>\n",
            "    import av\n",
            "ModuleNotFoundError: No module named 'av'\n",
            "\n",
            "Cannot import /content/drive/MyDrive/ComfyUI/comfy_extras/nodes_video.py module for custom nodes: No module named 'av'\n",
            "Traceback (most recent call last):\n",
            "  File \"/content/drive/MyDrive/ComfyUI/nodes.py\", line 2128, in load_custom_node\n",
            "    module_spec.loader.exec_module(module)\n",
            "  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n",
            "  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_ideogram.py\", line 21, in <module>\n",
            "    from comfy_api_nodes.apinode_utils import (\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/apinode_utils.py\", line 5, in <module>\n",
            "    from comfy_api.input_impl import VideoFromFile\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/__init__.py\", line 1, in <module>\n",
            "    from .video_types import VideoFromFile, VideoFromComponents\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/video_types.py\", line 2, in <module>\n",
            "    from av.container import InputContainer\n",
            "ModuleNotFoundError: No module named 'av'\n",
            "\n",
            "Cannot import /content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_ideogram.py module for custom nodes: No module named 'av'\n",
            "Traceback (most recent call last):\n",
            "  File \"/content/drive/MyDrive/ComfyUI/nodes.py\", line 2128, in load_custom_node\n",
            "    module_spec.loader.exec_module(module)\n",
            "  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n",
            "  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_openai.py\", line 22, in <module>\n",
            "    from comfy_api_nodes.apinode_utils import (\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/apinode_utils.py\", line 5, in <module>\n",
            "    from comfy_api.input_impl import VideoFromFile\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/__init__.py\", line 1, in <module>\n",
            "    from .video_types import VideoFromFile, VideoFromComponents\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/video_types.py\", line 2, in <module>\n",
            "    from av.container import InputContainer\n",
            "ModuleNotFoundError: No module named 'av'\n",
            "\n",
            "Cannot import /content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_openai.py module for custom nodes: No module named 'av'\n",
            "Traceback (most recent call last):\n",
            "  File \"/content/drive/MyDrive/ComfyUI/nodes.py\", line 2128, in load_custom_node\n",
            "    module_spec.loader.exec_module(module)\n",
            "  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n",
            "  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_minimax.py\", line 2, in <module>\n",
            "    from comfy_api.input_impl.video_types import VideoFromFile\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/__init__.py\", line 1, in <module>\n",
            "    from .video_types import VideoFromFile, VideoFromComponents\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/video_types.py\", line 2, in <module>\n",
            "    from av.container import InputContainer\n",
            "ModuleNotFoundError: No module named 'av'\n",
            "\n",
            "Cannot import /content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_minimax.py module for custom nodes: No module named 'av'\n",
            "Traceback (most recent call last):\n",
            "  File \"/content/drive/MyDrive/ComfyUI/nodes.py\", line 2128, in load_custom_node\n",
            "    module_spec.loader.exec_module(module)\n",
            "  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n",
            "  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_veo2.py\", line 8, in <module>\n",
            "    from comfy_api.input_impl.video_types import VideoFromFile\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/__init__.py\", line 1, in <module>\n",
            "    from .video_types import VideoFromFile, VideoFromComponents\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/video_types.py\", line 2, in <module>\n",
            "    from av.container import InputContainer\n",
            "ModuleNotFoundError: No module named 'av'\n",
            "\n",
            "Cannot import /content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_veo2.py module for custom nodes: No module named 'av'\n",
            "Traceback (most recent call last):\n",
            "  File \"/content/drive/MyDrive/ComfyUI/nodes.py\", line 2128, in load_custom_node\n",
            "    module_spec.loader.exec_module(module)\n",
            "  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n",
            "  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_kling.py\", line 59, in <module>\n",
            "    from comfy_api_nodes.apinode_utils import (\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/apinode_utils.py\", line 5, in <module>\n",
            "    from comfy_api.input_impl import VideoFromFile\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/__init__.py\", line 1, in <module>\n",
            "    from .video_types import VideoFromFile, VideoFromComponents\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/video_types.py\", line 2, in <module>\n",
            "    from av.container import InputContainer\n",
            "ModuleNotFoundError: No module named 'av'\n",
            "\n",
            "Cannot import /content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_kling.py module for custom nodes: No module named 'av'\n",
            "Traceback (most recent call last):\n",
            "  File \"/content/drive/MyDrive/ComfyUI/nodes.py\", line 2128, in load_custom_node\n",
            "    module_spec.loader.exec_module(module)\n",
            "  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n",
            "  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_bfl.py\", line 19, in <module>\n",
            "    from comfy_api_nodes.apinode_utils import (\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/apinode_utils.py\", line 5, in <module>\n",
            "    from comfy_api.input_impl import VideoFromFile\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/__init__.py\", line 1, in <module>\n",
            "    from .video_types import VideoFromFile, VideoFromComponents\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/video_types.py\", line 2, in <module>\n",
            "    from av.container import InputContainer\n",
            "ModuleNotFoundError: No module named 'av'\n",
            "\n",
            "Cannot import /content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_bfl.py module for custom nodes: No module named 'av'\n",
            "Traceback (most recent call last):\n",
            "  File \"/content/drive/MyDrive/ComfyUI/nodes.py\", line 2128, in load_custom_node\n",
            "    module_spec.loader.exec_module(module)\n",
            "  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n",
            "  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_luma.py\", line 3, in <module>\n",
            "    from comfy_api.input_impl.video_types import VideoFromFile\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/__init__.py\", line 1, in <module>\n",
            "    from .video_types import VideoFromFile, VideoFromComponents\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/video_types.py\", line 2, in <module>\n",
            "    from av.container import InputContainer\n",
            "ModuleNotFoundError: No module named 'av'\n",
            "\n",
            "Cannot import /content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_luma.py module for custom nodes: No module named 'av'\n",
            "Traceback (most recent call last):\n",
            "  File \"/content/drive/MyDrive/ComfyUI/nodes.py\", line 2128, in load_custom_node\n",
            "    module_spec.loader.exec_module(module)\n",
            "  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n",
            "  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_recraft.py\", line 24, in <module>\n",
            "    from comfy_api_nodes.apinode_utils import (\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/apinode_utils.py\", line 5, in <module>\n",
            "    from comfy_api.input_impl import VideoFromFile\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/__init__.py\", line 1, in <module>\n",
            "    from .video_types import VideoFromFile, VideoFromComponents\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/video_types.py\", line 2, in <module>\n",
            "    from av.container import InputContainer\n",
            "ModuleNotFoundError: No module named 'av'\n",
            "\n",
            "Cannot import /content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_recraft.py module for custom nodes: No module named 'av'\n",
            "Traceback (most recent call last):\n",
            "  File \"/content/drive/MyDrive/ComfyUI/nodes.py\", line 2128, in load_custom_node\n",
            "    module_spec.loader.exec_module(module)\n",
            "  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n",
            "  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_pixverse.py\", line 25, in <module>\n",
            "    from comfy_api_nodes.apinode_utils import (\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/apinode_utils.py\", line 5, in <module>\n",
            "    from comfy_api.input_impl import VideoFromFile\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/__init__.py\", line 1, in <module>\n",
            "    from .video_types import VideoFromFile, VideoFromComponents\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/video_types.py\", line 2, in <module>\n",
            "    from av.container import InputContainer\n",
            "ModuleNotFoundError: No module named 'av'\n",
            "\n",
            "Cannot import /content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_pixverse.py module for custom nodes: No module named 'av'\n",
            "Traceback (most recent call last):\n",
            "  File \"/content/drive/MyDrive/ComfyUI/nodes.py\", line 2128, in load_custom_node\n",
            "    module_spec.loader.exec_module(module)\n",
            "  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n",
            "  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_stability.py\", line 23, in <module>\n",
            "    from comfy_api_nodes.apinode_utils import (\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/apinode_utils.py\", line 5, in <module>\n",
            "    from comfy_api.input_impl import VideoFromFile\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/__init__.py\", line 1, in <module>\n",
            "    from .video_types import VideoFromFile, VideoFromComponents\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/video_types.py\", line 2, in <module>\n",
            "    from av.container import InputContainer\n",
            "ModuleNotFoundError: No module named 'av'\n",
            "\n",
            "Cannot import /content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_stability.py module for custom nodes: No module named 'av'\n",
            "Traceback (most recent call last):\n",
            "  File \"/content/drive/MyDrive/ComfyUI/nodes.py\", line 2128, in load_custom_node\n",
            "    module_spec.loader.exec_module(module)\n",
            "  File \"<frozen importlib._bootstrap_external>\", line 940, in exec_module\n",
            "  File \"<frozen importlib._bootstrap>\", line 241, in _call_with_frames_removed\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_pika.py\", line 34, in <module>\n",
            "    from comfy_api_nodes.apinode_utils import (\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api_nodes/apinode_utils.py\", line 5, in <module>\n",
            "    from comfy_api.input_impl import VideoFromFile\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/__init__.py\", line 1, in <module>\n",
            "    from .video_types import VideoFromFile, VideoFromComponents\n",
            "  File \"/content/drive/MyDrive/ComfyUI/comfy_api/input_impl/video_types.py\", line 2, in <module>\n",
            "    from av.container import InputContainer\n",
            "ModuleNotFoundError: No module named 'av'\n",
            "\n",
            "Cannot import /content/drive/MyDrive/ComfyUI/comfy_api_nodes/nodes_pika.py module for custom nodes: No module named 'av'\n",
            "### Loading: ComfyUI-Manager (V3.31.13)\n",
            "[ComfyUI-Manager] network_mode: public\n",
            "### ComfyUI Version: v0.3.32-2-g271c9c5b | Released on '2025-05-06'\n",
            "[ComfyUI-Manager] default cache updated: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/model-list.json\n",
            "[ComfyUI-Manager] default cache updated: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/alter-list.json\n",
            "[ComfyUI-Manager] default cache updated: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/extension-node-map.json\n",
            "[ComfyUI-Manager] default cache updated: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/github-stats.json\n",
            "----------Jake Upgrade Nodes Loaded----------\n",
            "[ComfyUI-Manager] default cache updated: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/custom-node-list.json\n",
            "\n",
            "Import times for custom nodes:\n",
            "   0.0 seconds: /content/drive/MyDrive/ComfyUI/custom_nodes/websocket_image_save.py\n",
            "   0.5 seconds: /content/drive/MyDrive/ComfyUI/custom_nodes/comfyui-jakeupgrade\n",
            "   0.6 seconds: /content/drive/MyDrive/ComfyUI/custom_nodes/ComfyUI-Manager\n",
            "\n",
            "WARNING: some comfy_api_nodes/ nodes did not import correctly. This may be because they are missing some dependencies.\n",
            "\n",
            "IMPORT FAILED: nodes_ideogram.py\n",
            "IMPORT FAILED: nodes_openai.py\n",
            "IMPORT FAILED: nodes_minimax.py\n",
            "IMPORT FAILED: nodes_veo2.py\n",
            "IMPORT FAILED: nodes_kling.py\n",
            "IMPORT FAILED: nodes_bfl.py\n",
            "IMPORT FAILED: nodes_luma.py\n",
            "IMPORT FAILED: nodes_recraft.py\n",
            "IMPORT FAILED: nodes_pixverse.py\n",
            "IMPORT FAILED: nodes_stability.py\n",
            "IMPORT FAILED: nodes_pika.py\n",
            "\n",
            "This issue might be caused by new missing dependencies added the last time you updated ComfyUI.\n",
            "Please do a: pip install -r requirements.txt\n",
            "\n",
            "WARNING: some comfy_extras/ nodes did not import correctly. This may be because they are missing some dependencies.\n",
            "\n",
            "IMPORT FAILED: nodes_lt.py\n",
            "IMPORT FAILED: nodes_video.py\n",
            "\n",
            "This issue might be caused by new missing dependencies added the last time you updated ComfyUI.\n",
            "Please do a: pip install -r requirements.txt\n",
            "\n",
            "********** ERROR ***********\n",
            "\n",
            "comfyui-workflow-templates is not installed.\n",
            "\n",
            "Please install the updated requirements.txt file by running:\n",
            "/usr/bin/python3 -m pip install -r /content/drive/MyDrive/ComfyUI/requirements.txt\n",
            "\n",
            "This error is happening because the ComfyUI frontend is no longer shipped as part of the main repo but as a pip package instead.\n",
            "\n",
            "If you are on the portable package you can run: update\\update_comfyui.bat to solve this problem\n",
            "\n",
            "********** ERROR ***********\n",
            "FETCH ComfyRegistry Data: 5/84\n",
            "FETCH ComfyRegistry Data: 10/84\n",
            "FETCH ComfyRegistry Data: 15/84\n",
            "FETCH ComfyRegistry Data: 20/84\n",
            "FETCH ComfyRegistry Data: 25/84\n",
            "FETCH ComfyRegistry Data: 30/84\n",
            "FETCH ComfyRegistry Data: 35/84\n",
            "FETCH ComfyRegistry Data: 40/84\n",
            "FETCH ComfyRegistry Data: 45/84\n",
            "FETCH ComfyRegistry Data: 50/84\n"
          ]
        }
      ],
      "source": [
        "!wget -P ~ https://github.com/cloudflare/cloudflared/releases/latest/download/cloudflared-linux-amd64.deb\n",
        "!dpkg -i ~/cloudflared-linux-amd64.deb\n",
        "\n",
        "import subprocess\n",
        "import threading\n",
        "import time\n",
        "import socket\n",
        "import urllib.request\n",
        "\n",
        "def iframe_thread(port):\n",
        "  while True:\n",
        "      time.sleep(0.5)\n",
        "      sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)\n",
        "      result = sock.connect_ex(('127.0.0.1', port))\n",
        "      if result == 0:\n",
        "        break\n",
        "      sock.close()\n",
        "  print(\"\\nComfyUI finished loading, trying to launch cloudflared (if it gets stuck here cloudflared is having issues)\\n\")\n",
        "\n",
        "  p = subprocess.Popen([\"cloudflared\", \"tunnel\", \"--url\", \"http://127.0.0.1:{}\".format(port)], stdout=subprocess.PIPE, stderr=subprocess.PIPE)\n",
        "  for line in p.stderr:\n",
        "    l = line.decode()\n",
        "    if \"trycloudflare.com \" in l:\n",
        "      print(\"This is the URL to access ComfyUI:\", l[l.find(\"http\"):], end='')\n",
        "    #print(l, end='')\n",
        "\n",
        "\n",
        "threading.Thread(target=iframe_thread, daemon=True, args=(8188,)).start()\n",
        "\n",
        "!python main.py --dont-print-server"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "kkkkkkkkkkkkkk"
      },
      "source": [
        "### Run ComfyUI with localtunnel\n",
        "\n",
        "\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "jjjjjjjjjjjjj"
      },
      "outputs": [],
      "source": [
        "!npm install -g localtunnel\n",
        "\n",
        "import subprocess\n",
        "import threading\n",
        "import time\n",
        "import socket\n",
        "import urllib.request\n",
        "\n",
        "def iframe_thread(port):\n",
        "  while True:\n",
        "      time.sleep(0.5)\n",
        "      sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)\n",
        "      result = sock.connect_ex(('127.0.0.1', port))\n",
        "      if result == 0:\n",
        "        break\n",
        "      sock.close()\n",
        "  print(\"\\nComfyUI finished loading, trying to launch localtunnel (if it gets stuck here localtunnel is having issues)\\n\")\n",
        "\n",
        "  print(\"The password/enpoint ip for localtunnel is:\", urllib.request.urlopen('https://ipv4.icanhazip.com').read().decode('utf8').strip(\"\\n\"))\n",
        "  p = subprocess.Popen([\"lt\", \"--port\", \"{}\".format(port)], stdout=subprocess.PIPE)\n",
        "  for line in p.stdout:\n",
        "    print(line.decode(), end='')\n",
        "\n",
        "\n",
        "threading.Thread(target=iframe_thread, daemon=True, args=(8188,)).start()\n",
        "\n",
        "!python main.py --dont-print-server"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "gggggggggg"
      },
      "source": [
        "### Run ComfyUI with colab iframe (use only in case the previous way with localtunnel doesn't work)\n",
        "\n",
        "You should see the ui appear in an iframe. If you get a 403 error, it's your firefox settings or an extension that's messing things up.\n",
        "\n",
        "If you want to open it in another window use the link.\n",
        "\n",
        "Note that some UI features like live image previews won't work because the colab iframe blocks websockets."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "hhhhhhhhhh"
      },
      "outputs": [],
      "source": [
        "import threading\n",
        "import time\n",
        "import socket\n",
        "def iframe_thread(port):\n",
        "  while True:\n",
        "      time.sleep(0.5)\n",
        "      sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)\n",
        "      result = sock.connect_ex(('127.0.0.1', port))\n",
        "      if result == 0:\n",
        "        break\n",
        "      sock.close()\n",
        "  from google.colab import output\n",
        "  output.serve_kernel_port_as_iframe(port, height=1024)\n",
        "  print(\"to open it in a window you can open this link here:\")\n",
        "  output.serve_kernel_port_as_window(port)\n",
        "\n",
        "threading.Thread(target=iframe_thread, daemon=True, args=(8188,)).start()\n",
        "\n",
        "!python main.py --dont-print-server"
      ]
    }
  ],
  "metadata": {
    "accelerator": "GPU",
    "colab": {
      "provenance": []
    },
    "gpuClass": "standard",
    "kernelspec": {
      "display_name": "Python 3",
      "name": "python3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 0
}